// intl-tel-input/svelte/build/IntlTelInput.mjs
var $n = Object.defineProperty;
var Re = (t) => {
  throw TypeError(t);
};
var Kn = (t, e, n) => e in t ? $n(t, e, { enumerable: !0, configurable: !0, writable: !0, value: n }) : t[e] = n;
var et = (t, e, n) => Kn(t, typeof e != "symbol" ? e + "" : e, n), ie = (t, e, n) => e.has(t) || Re("Cannot " + n);
var A = (t, e, n) => (ie(t, e, "read from private field"), n ? n.call(t) : e.get(t)), X = (t, e, n) => e.has(t) ? Re("Cannot add the same private member more than once") : e instanceof WeakSet ? e.add(t) : e.set(t, n), nt = (t, e, n, i) => (ie(t, e, "write to private field"), i ? i.call(t, n) : e.set(t, n), n), vt = (t, e, n) => (ie(t, e, "access private method"), n);
import { onMount as zn, onDestroy as qn } from "svelte";
const Xn = "5";
var nn;
typeof window < "u" && ((nn = window.__svelte ?? (window.__svelte = {})).v ?? (nn.v = /* @__PURE__ */ new Set())).add(Xn);
const Qn = 1, Zn = 4, Jn = 8, ti = 16, ei = 2, P = Symbol(), ni = "http://www.w3.org/1999/xhtml", ii = "@attach", fe = !1;
var sn = Array.isArray, si = Array.prototype.indexOf, ri = Object.defineProperty, $t = Object.getOwnPropertyDescriptor, oi = Object.getOwnPropertyDescriptors, li = Object.prototype, ai = Array.prototype, rn = Object.getPrototypeOf;
function ui(t) {
  for (var e = 0; e < t.length; e++)
    t[e]();
}
function on() {
  var t, e, n = new Promise((i, s) => {
    t = i, e = s;
  });
  return { promise: n, resolve: t, reject: e };
}
const k = 2, qt = 4, be = 8, we = 1 << 24, yt = 16, mt = 32, Ot = 64, ln = 128, $ = 512, U = 1024, j = 2048, lt = 4096, Et = 8192, tt = 16384, Ie = 32768, ke = 65536, Me = 1 << 17, an = 1 << 18, ne = 1 << 19, ci = 1 << 20, _t = 32768, he = 1 << 21, Ee = 1 << 22, it = 1 << 23, ft = Symbol("$state"), di = Symbol("legacy props"), fi = Symbol(""), bt = new class extends Error {
  constructor() {
    super(...arguments);
    et(this, "name", "StaleReactionError");
    et(this, "message", "The reaction that called `getAbortSignal()` was re-run or destroyed");
  }
}();
function hi() {
  throw new Error("https://svelte.dev/e/async_derived_orphan");
}
function pi(t) {
  throw new Error("https://svelte.dev/e/effect_in_teardown");
}
function _i() {
  throw new Error("https://svelte.dev/e/effect_in_unowned_derived");
}
function Ci(t) {
  throw new Error("https://svelte.dev/e/effect_orphan");
}
function yi() {
  throw new Error("https://svelte.dev/e/effect_update_depth_exceeded");
}
function mi(t) {
  throw new Error("https://svelte.dev/e/props_invalid_value");
}
function gi() {
  throw new Error("https://svelte.dev/e/state_descriptors_fixed");
}
function vi() {
  throw new Error("https://svelte.dev/e/state_prototype_fixed");
}
function bi() {
  throw new Error("https://svelte.dev/e/state_unsafe_mutation");
}
function wi() {
  console.warn("https://svelte.dev/e/select_multiple_invalid_value");
}
function un(t) {
  return t === this.v;
}
function Ii(t, e) {
  return t != t ? e == e : t !== e || t !== null && typeof t == "object" || typeof t == "function";
}
function Ei(t) {
  return !Ii(t, this.v);
}
let Ni = !1, Q = null;
function Xt(t) {
  Q = t;
}
function Di(t, e = !1, n) {
  Q = {
    p: Q,
    i: !1,
    c: null,
    e: null,
    s: t,
    x: null,
    l: null
  };
}
function Li(t) {
  var e = (
    /** @type {ComponentContext} */
    Q
  ), n = e.e;
  if (n !== null) {
    e.e = null;
    for (var i of n)
      wn(i);
  }
  return t !== void 0 && (e.x = t), e.i = !0, Q = e.p, t ?? /** @type {T} */
  {};
}
function cn() {
  return !0;
}
let wt = [];
function Ai() {
  var t = wt;
  wt = [], ui(t);
}
function xt(t) {
  if (wt.length === 0) {
    var e = wt;
    queueMicrotask(() => {
      e === wt && Ai();
    });
  }
  wt.push(t);
}
function Ti(t) {
  var e = D;
  if (e === null)
    return w.f |= it, t;
  if ((e.f & Ie) === 0) {
    if ((e.f & ln) === 0)
      throw t;
    e.b.error(t);
  } else
    Qt(t, e);
}
function Qt(t, e) {
  for (; e !== null; ) {
    if ((e.f & ln) !== 0)
      try {
        e.b.error(t);
        return;
      } catch (n) {
        t = n;
      }
    e = e.parent;
  }
  throw t;
}
const Si = -7169;
function F(t, e) {
  t.f = t.f & Si | e;
}
function Ne(t) {
  (t.f & $) !== 0 || t.deps === null ? F(t, U) : F(t, lt);
}
function dn(t) {
  if (t !== null)
    for (const e of t)
      (e.f & k) === 0 || (e.f & _t) === 0 || (e.f ^= _t, dn(
        /** @type {Derived} */
        e.deps
      ));
}
function Pi(t, e, n) {
  (t.f & j) !== 0 ? e.add(t) : (t.f & lt) !== 0 && n.add(t), dn(t.deps), F(t, U);
}
const Vt = /* @__PURE__ */ new Set();
let T = null, R = null, z = [], De = null, pe = !1;
var Nt, Dt, dt, Lt, Ht, At, Tt, St, J, _e, Ce, fn;
const Oe = class Oe {
  constructor() {
    X(this, J);
    et(this, "committed", !1);
    /**
     * The current values of any sources that are updated in this batch
     * They keys of this map are identical to `this.#previous`
     * @type {Map<Source, any>}
     */
    et(this, "current", /* @__PURE__ */ new Map());
    /**
     * The values of any sources that are updated in this batch _before_ those updates took place.
     * They keys of this map are identical to `this.#current`
     * @type {Map<Source, any>}
     */
    et(this, "previous", /* @__PURE__ */ new Map());
    /**
     * When the batch is committed (and the DOM is updated), we need to remove old branches
     * and append new ones by calling the functions added inside (if/each/key/etc) blocks
     * @type {Set<() => void>}
     */
    X(this, Nt, /* @__PURE__ */ new Set());
    /**
     * If a fork is discarded, we need to destroy any effects that are no longer needed
     * @type {Set<(batch: Batch) => void>}
     */
    X(this, Dt, /* @__PURE__ */ new Set());
    /**
     * The number of async effects that are currently in flight
     */
    X(this, dt, 0);
    /**
     * The number of async effects that are currently in flight, _not_ inside a pending boundary
     */
    X(this, Lt, 0);
    /**
     * A deferred that resolves when the batch is committed, used with `settled()`
     * TODO replace with Promise.withResolvers once supported widely enough
     * @type {{ promise: Promise<void>, resolve: (value?: any) => void, reject: (reason: unknown) => void } | null}
     */
    X(this, Ht, null);
    /**
     * Deferred effects (which run after async work has completed) that are DIRTY
     * @type {Set<Effect>}
     */
    X(this, At, /* @__PURE__ */ new Set());
    /**
     * Deferred effects that are MAYBE_DIRTY
     * @type {Set<Effect>}
     */
    X(this, Tt, /* @__PURE__ */ new Set());
    /**
     * A set of branches that still exist, but will be destroyed when this batch
     * is committed — we skip over these during `process`
     * @type {Set<Effect>}
     */
    et(this, "skipped_effects", /* @__PURE__ */ new Set());
    et(this, "is_fork", !1);
    X(this, St, !1);
  }
  is_deferred() {
    return this.is_fork || A(this, Lt) > 0;
  }
  /**
   *
   * @param {Effect[]} root_effects
   */
  process(e) {
    var s;
    z = [], this.apply();
    var n = [], i = [];
    for (const r of e)
      vt(this, J, _e).call(this, r, n, i);
    if (this.is_deferred())
      vt(this, J, Ce).call(this, i), vt(this, J, Ce).call(this, n);
    else {
      for (const r of A(this, Nt)) r();
      A(this, Nt).clear(), A(this, dt) === 0 && vt(this, J, fn).call(this), T = null, Ue(i), Ue(n), (s = A(this, Ht)) == null || s.resolve();
    }
    R = null;
  }
  /**
   * Associate a change to a given source with the current
   * batch, noting its previous and current values
   * @param {Source} source
   * @param {any} value
   */
  capture(e, n) {
    n !== P && !this.previous.has(e) && this.previous.set(e, n), (e.f & it) === 0 && (this.current.set(e, e.v), R == null || R.set(e, e.v));
  }
  activate() {
    T = this, this.apply();
  }
  deactivate() {
    T === this && (T = null, R = null);
  }
  flush() {
    if (this.activate(), z.length > 0) {
      if (Oi(), T !== null && T !== this)
        return;
    } else A(this, dt) === 0 && this.process([]);
    this.deactivate();
  }
  discard() {
    for (const e of A(this, Dt)) e(this);
    A(this, Dt).clear();
  }
  /**
   *
   * @param {boolean} blocking
   */
  increment(e) {
    nt(this, dt, A(this, dt) + 1), e && nt(this, Lt, A(this, Lt) + 1);
  }
  /**
   *
   * @param {boolean} blocking
   */
  decrement(e) {
    nt(this, dt, A(this, dt) - 1), e && nt(this, Lt, A(this, Lt) - 1), !A(this, St) && (nt(this, St, !0), xt(() => {
      nt(this, St, !1), this.is_deferred() ? z.length > 0 && this.flush() : this.revive();
    }));
  }
  revive() {
    for (const e of A(this, At))
      A(this, Tt).delete(e), F(e, j), Pt(e);
    for (const e of A(this, Tt))
      F(e, lt), Pt(e);
    this.flush();
  }
  /** @param {() => void} fn */
  oncommit(e) {
    A(this, Nt).add(e);
  }
  /** @param {(batch: Batch) => void} fn */
  ondiscard(e) {
    A(this, Dt).add(e);
  }
  settled() {
    return (A(this, Ht) ?? nt(this, Ht, on())).promise;
  }
  static ensure() {
    if (T === null) {
      const e = T = new Oe();
      Vt.add(T), xt(() => {
        T === e && e.flush();
      });
    }
    return T;
  }
  apply() {
  }
};
Nt = new WeakMap(), Dt = new WeakMap(), dt = new WeakMap(), Lt = new WeakMap(), Ht = new WeakMap(), At = new WeakMap(), Tt = new WeakMap(), St = new WeakMap(), J = new WeakSet(), /**
 * Traverse the effect tree, executing effects or stashing
 * them for later execution as appropriate
 * @param {Effect} root
 * @param {Effect[]} effects
 * @param {Effect[]} render_effects
 */
_e = function(e, n, i) {
  e.f ^= U;
  for (var s = e.first, r = null; s !== null; ) {
    var o = s.f, l = (o & (mt | Ot)) !== 0, a = l && (o & U) !== 0, u = a || (o & Et) !== 0 || this.skipped_effects.has(s);
    if (!u && s.fn !== null) {
      l ? s.f ^= U : r !== null && (o & (qt | be | we)) !== 0 ? r.b.defer_effect(s) : (o & qt) !== 0 ? n.push(s) : Bt(s) && ((o & yt) !== 0 && A(this, At).add(s), Ft(s));
      var c = s.first;
      if (c !== null) {
        s = c;
        continue;
      }
    }
    var d = s.parent;
    for (s = s.next; s === null && d !== null; )
      d === r && (r = null), s = d.next, d = d.parent;
  }
}, /**
 * @param {Effect[]} effects
 */
Ce = function(e) {
  for (var n = 0; n < e.length; n += 1)
    Pi(e[n], A(this, At), A(this, Tt));
}, fn = function() {
  var s;
  if (Vt.size > 1) {
    this.previous.clear();
    var e = R, n = !0;
    for (const r of Vt) {
      if (r === this) {
        n = !1;
        continue;
      }
      const o = [];
      for (const [a, u] of this.current) {
        if (r.current.has(a))
          if (n && u !== r.current.get(a))
            r.current.set(a, u);
          else
            continue;
        o.push(a);
      }
      if (o.length === 0)
        continue;
      const l = [...r.current.keys()].filter((a) => !this.current.has(a));
      if (l.length > 0) {
        var i = z;
        z = [];
        const a = /* @__PURE__ */ new Set(), u = /* @__PURE__ */ new Map();
        for (const c of o)
          hn(c, l, a, u);
        if (z.length > 0) {
          T = r, r.apply();
          for (const c of z)
            vt(s = r, J, _e).call(s, c, [], []);
          r.deactivate();
        }
        z = i;
      }
    }
    T = null, R = e;
  }
  this.committed = !0, Vt.delete(this);
};
let Zt = Oe;
function Oi() {
  pe = !0;
  var t = null;
  try {
    for (var e = 0; z.length > 0; ) {
      var n = Zt.ensure();
      if (e++ > 1e3) {
        var i, s;
        Ri();
      }
      n.process(z), st.clear();
    }
  } finally {
    pe = !1, De = null;
  }
}
function Ri() {
  try {
    yi();
  } catch (t) {
    Qt(t, De);
  }
}
let K = null;
function Ue(t) {
  var e = t.length;
  if (e !== 0) {
    for (var n = 0; n < e; ) {
      var i = t[n++];
      if ((i.f & (tt | Et)) === 0 && Bt(i) && (K = /* @__PURE__ */ new Set(), Ft(i), i.deps === null && i.first === null && i.nodes === null && (i.teardown === null && i.ac === null ? Ln(i) : i.fn = null), (K == null ? void 0 : K.size) > 0)) {
        st.clear();
        for (const s of K) {
          if ((s.f & (tt | Et)) !== 0) continue;
          const r = [s];
          let o = s.parent;
          for (; o !== null; )
            K.has(o) && (K.delete(o), r.push(o)), o = o.parent;
          for (let l = r.length - 1; l >= 0; l--) {
            const a = r[l];
            (a.f & (tt | Et)) === 0 && Ft(a);
          }
        }
        K.clear();
      }
    }
    K = null;
  }
}
function hn(t, e, n, i) {
  if (!n.has(t) && (n.add(t), t.reactions !== null))
    for (const s of t.reactions) {
      const r = s.f;
      (r & k) !== 0 ? hn(
        /** @type {Derived} */
        s,
        e,
        n,
        i
      ) : (r & (Ee | yt)) !== 0 && (r & j) === 0 && pn(s, e, i) && (F(s, j), Pt(
        /** @type {Effect} */
        s
      ));
    }
}
function pn(t, e, n) {
  const i = n.get(t);
  if (i !== void 0) return i;
  if (t.deps !== null)
    for (const s of t.deps) {
      if (e.includes(s))
        return !0;
      if ((s.f & k) !== 0 && pn(
        /** @type {Derived} */
        s,
        e,
        n
      ))
        return n.set(
          /** @type {Derived} */
          s,
          !0
        ), !0;
    }
  return n.set(t, !1), !1;
}
function Pt(t) {
  for (var e = De = t; e.parent !== null; ) {
    e = e.parent;
    var n = e.f;
    if (pe && e === D && (n & yt) !== 0 && (n & an) === 0)
      return;
    if ((n & (Ot | mt)) !== 0) {
      if ((n & U) === 0) return;
      e.f ^= U;
    }
  }
  z.push(e);
}
function ki(t, e, n, i) {
  const s = Le;
  var r = t.filter((f) => !f.settled);
  if (n.length === 0 && r.length === 0) {
    i(e.map(s));
    return;
  }
  var o = T, l = (
    /** @type {Effect} */
    D
  ), a = Mi(), u = r.length === 1 ? r[0].promise : r.length > 1 ? Promise.all(r.map((f) => f.promise)) : null;
  function c(f) {
    a();
    try {
      i(f);
    } catch (h) {
      (l.f & tt) === 0 && Qt(h, l);
    }
    o == null || o.deactivate(), ye();
  }
  if (n.length === 0) {
    u.then(() => c(e.map(s)));
    return;
  }
  function d() {
    a(), Promise.all(n.map((f) => /* @__PURE__ */ Ui(f))).then((f) => c([...e.map(s), ...f])).catch((f) => Qt(f, l));
  }
  u ? u.then(d) : d();
}
function Mi() {
  var t = D, e = w, n = Q, i = T;
  return function(r = !0) {
    ot(t), Z(e), Xt(n), r && (i == null || i.activate());
  };
}
function ye() {
  ot(null), Z(null), Xt(null);
}
// @__NO_SIDE_EFFECTS__
function Le(t) {
  var e = k | j, n = w !== null && (w.f & k) !== 0 ? (
    /** @type {Derived} */
    w
  ) : null;
  return D !== null && (D.f |= ne), {
    ctx: Q,
    deps: null,
    effects: null,
    equals: un,
    f: e,
    fn: t,
    reactions: null,
    rv: 0,
    v: (
      /** @type {V} */
      P
    ),
    wv: 0,
    parent: n ?? D,
    ac: null
  };
}
// @__NO_SIDE_EFFECTS__
function Ui(t, e, n) {
  let i = (
    /** @type {Effect | null} */
    D
  );
  i === null && hi();
  var s = (
    /** @type {Boundary} */
    i.b
  ), r = (
    /** @type {Promise<V>} */
    /** @type {unknown} */
    void 0
  ), o = mn(
    /** @type {V} */
    P
  ), l = !w, a = /* @__PURE__ */ new Map();
  return Xi(() => {
    var h;
    var u = on();
    r = u.promise;
    try {
      Promise.resolve(t()).then(u.resolve, u.reject).then(() => {
        c === T && c.committed && c.deactivate(), ye();
      });
    } catch (_) {
      u.reject(_), ye();
    }
    var c = (
      /** @type {Batch} */
      T
    );
    if (l) {
      var d = s.is_rendered();
      s.update_pending_count(1), c.increment(d), (h = a.get(c)) == null || h.reject(bt), a.delete(c), a.set(c, u);
    }
    const f = (_, g = void 0) => {
      if (c.activate(), g)
        g !== bt && (o.f |= it, ge(o, g));
      else {
        (o.f & it) !== 0 && (o.f ^= it), ge(o, _);
        for (const [b, y] of a) {
          if (a.delete(b), b === c) break;
          y.reject(bt);
        }
      }
      l && (s.update_pending_count(-1), c.decrement(d));
    };
    u.promise.then(f, (_) => f(null, _ || "unknown"));
  }), bn(() => {
    for (const u of a.values())
      u.reject(bt);
  }), new Promise((u) => {
    function c(d) {
      function f() {
        d === r ? u(o) : c(r);
      }
      d.then(f, f);
    }
    c(r);
  });
}
// @__NO_SIDE_EFFECTS__
function xi(t) {
  const e = /* @__PURE__ */ Le(t);
  return e.equals = Ei, e;
}
function _n(t) {
  var e = t.effects;
  if (e !== null) {
    t.effects = null;
    for (var n = 0; n < e.length; n += 1)
      Ct(
        /** @type {Effect} */
        e[n]
      );
  }
}
function Fi(t) {
  for (var e = t.parent; e !== null; ) {
    if ((e.f & k) === 0)
      return (e.f & tt) === 0 ? (
        /** @type {Effect} */
        e
      ) : null;
    e = e.parent;
  }
  return null;
}
function Ae(t) {
  var e, n = D;
  ot(Fi(t));
  try {
    t.f &= ~_t, _n(t), e = Pn(t);
  } finally {
    ot(n);
  }
  return e;
}
function Cn(t) {
  var e = Ae(t);
  if (!t.equals(e) && (t.wv = Tn(), (!(T != null && T.is_fork) || t.deps === null) && (t.v = e, t.deps === null))) {
    F(t, U);
    return;
  }
  rt || (R !== null ? (vn() || T != null && T.is_fork) && R.set(t, e) : Ne(t));
}
let me = /* @__PURE__ */ new Set();
const st = /* @__PURE__ */ new Map();
let yn = !1;
function mn(t, e) {
  var n = {
    f: 0,
    // TODO ideally we could skip this altogether, but it causes type errors
    v: t,
    reactions: null,
    equals: un,
    rv: 0,
    wv: 0
  };
  return n;
}
// @__NO_SIDE_EFFECTS__
function Y(t, e) {
  const n = mn(t);
  return ts(n), n;
}
function W(t, e, n = !1) {
  w !== null && // since we are untracking the function inside `$inspect.with` we need to add this check
  // to ensure we error if state is set inside an inspect effect
  (!q || (w.f & Me) !== 0) && cn() && (w.f & (k | yt | Ee | Me)) !== 0 && !(x != null && x.includes(t)) && bi();
  let i = n ? It(e) : e;
  return ge(t, i);
}
function ge(t, e) {
  if (!t.equals(e)) {
    var n = t.v;
    rt ? st.set(t, e) : st.set(t, n), t.v = e;
    var i = Zt.ensure();
    if (i.capture(t, n), (t.f & k) !== 0) {
      const s = (
        /** @type {Derived} */
        t
      );
      (t.f & j) !== 0 && Ae(s), Ne(s);
    }
    t.wv = Tn(), gn(t, j), D !== null && (D.f & U) !== 0 && (D.f & (mt | Ot)) === 0 && (G === null ? es([t]) : G.push(t)), !i.is_fork && me.size > 0 && !yn && Hi();
  }
  return e;
}
function Hi() {
  yn = !1;
  for (const t of me)
    (t.f & U) !== 0 && F(t, lt), Bt(t) && Ft(t);
  me.clear();
}
function se(t) {
  W(t, t.v + 1);
}
function gn(t, e) {
  var n = t.reactions;
  if (n !== null)
    for (var i = n.length, s = 0; s < i; s++) {
      var r = n[s], o = r.f, l = (o & j) === 0;
      if (l && F(r, e), (o & k) !== 0) {
        var a = (
          /** @type {Derived} */
          r
        );
        R == null || R.delete(a), (o & _t) === 0 && (o & $ && (r.f |= _t), gn(a, lt));
      } else l && ((o & yt) !== 0 && K !== null && K.add(
        /** @type {Effect} */
        r
      ), Pt(
        /** @type {Effect} */
        r
      ));
    }
}
function It(t) {
  if (typeof t != "object" || t === null || ft in t)
    return t;
  const e = rn(t);
  if (e !== li && e !== ai)
    return t;
  var n = /* @__PURE__ */ new Map(), i = sn(t), s = /* @__PURE__ */ Y(0), r = ht, o = (l) => {
    if (ht === r)
      return l();
    var a = w, u = ht;
    Z(null), He(r);
    var c = l();
    return Z(a), He(u), c;
  };
  return i && n.set("length", /* @__PURE__ */ Y(
    /** @type {any[]} */
    t.length
  )), new Proxy(
    /** @type {any} */
    t,
    {
      defineProperty(l, a, u) {
        (!("value" in u) || u.configurable === !1 || u.enumerable === !1 || u.writable === !1) && gi();
        var c = n.get(a);
        return c === void 0 ? c = o(() => {
          var d = /* @__PURE__ */ Y(u.value);
          return n.set(a, d), d;
        }) : W(c, u.value, !0), !0;
      },
      deleteProperty(l, a) {
        var u = n.get(a);
        if (u === void 0) {
          if (a in l) {
            const c = o(() => /* @__PURE__ */ Y(P));
            n.set(a, c), se(s);
          }
        } else
          W(u, P), se(s);
        return !0;
      },
      get(l, a, u) {
        var h;
        if (a === ft)
          return t;
        var c = n.get(a), d = a in l;
        if (c === void 0 && (!d || (h = $t(l, a)) != null && h.writable) && (c = o(() => {
          var _ = It(d ? l[a] : P), g = /* @__PURE__ */ Y(_);
          return g;
        }), n.set(a, c)), c !== void 0) {
          var f = E(c);
          return f === P ? void 0 : f;
        }
        return Reflect.get(l, a, u);
      },
      getOwnPropertyDescriptor(l, a) {
        var u = Reflect.getOwnPropertyDescriptor(l, a);
        if (u && "value" in u) {
          var c = n.get(a);
          c && (u.value = E(c));
        } else if (u === void 0) {
          var d = n.get(a), f = d == null ? void 0 : d.v;
          if (d !== void 0 && f !== P)
            return {
              enumerable: !0,
              configurable: !0,
              value: f,
              writable: !0
            };
        }
        return u;
      },
      has(l, a) {
        var f;
        if (a === ft)
          return !0;
        var u = n.get(a), c = u !== void 0 && u.v !== P || Reflect.has(l, a);
        if (u !== void 0 || D !== null && (!c || (f = $t(l, a)) != null && f.writable)) {
          u === void 0 && (u = o(() => {
            var h = c ? It(l[a]) : P, _ = /* @__PURE__ */ Y(h);
            return _;
          }), n.set(a, u));
          var d = E(u);
          if (d === P)
            return !1;
        }
        return c;
      },
      set(l, a, u, c) {
        var p;
        var d = n.get(a), f = a in l;
        if (i && a === "length")
          for (var h = u; h < /** @type {Source<number>} */
          d.v; h += 1) {
            var _ = n.get(h + "");
            _ !== void 0 ? W(_, P) : h in l && (_ = o(() => /* @__PURE__ */ Y(P)), n.set(h + "", _));
          }
        if (d === void 0)
          (!f || (p = $t(l, a)) != null && p.writable) && (d = o(() => /* @__PURE__ */ Y(void 0)), W(d, It(u)), n.set(a, d));
        else {
          f = d.v !== P;
          var g = o(() => It(u));
          W(d, g);
        }
        var b = Reflect.getOwnPropertyDescriptor(l, a);
        if (b != null && b.set && b.set.call(c, u), !f) {
          if (i && typeof a == "string") {
            var y = (
              /** @type {Source<number>} */
              n.get("length")
            ), m = Number(a);
            Number.isInteger(m) && m >= y.v && W(y, m + 1);
          }
          se(s);
        }
        return !0;
      },
      ownKeys(l) {
        E(s);
        var a = Reflect.ownKeys(l).filter((d) => {
          var f = n.get(d);
          return f === void 0 || f.v !== P;
        });
        for (var [u, c] of n)
          c.v !== P && !(u in l) && a.push(u);
        return a;
      },
      setPrototypeOf() {
        vi();
      }
    }
  );
}
function xe(t) {
  try {
    if (t !== null && typeof t == "object" && ft in t)
      return t[ft];
  } catch {
  }
  return t;
}
function Bi(t, e) {
  return Object.is(xe(t), xe(e));
}
var Vi, Wi, ji;
// @__NO_SIDE_EFFECTS__
function Gi(t) {
  return (
    /** @type {TemplateNode | null} */
    Wi.call(t)
  );
}
// @__NO_SIDE_EFFECTS__
function Yi(t) {
  return (
    /** @type {TemplateNode | null} */
    ji.call(t)
  );
}
function $i(t, e) {
  if (e) {
    const n = document.body;
    t.autofocus = !0, xt(() => {
      document.activeElement === n && t.focus();
    });
  }
}
function Te(t) {
  var e = w, n = D;
  Z(null), ot(null);
  try {
    return t();
  } finally {
    Z(e), ot(n);
  }
}
function Ki(t) {
  D === null && (w === null && Ci(), _i()), rt && pi();
}
function zi(t, e) {
  var n = e.last;
  n === null ? e.last = e.first = t : (n.next = t, t.prev = n, e.last = t);
}
function gt(t, e, n) {
  var i = D;
  i !== null && (i.f & Et) !== 0 && (t |= Et);
  var s = {
    ctx: Q,
    deps: null,
    nodes: null,
    f: t | j | $,
    first: null,
    fn: e,
    last: null,
    next: null,
    parent: i,
    b: i && i.b,
    prev: null,
    teardown: null,
    wv: 0,
    ac: null
  };
  if (n)
    try {
      Ft(s), s.f |= Ie;
    } catch (l) {
      throw Ct(s), l;
    }
  else e !== null && Pt(s);
  var r = s;
  if (n && r.deps === null && r.teardown === null && r.nodes === null && r.first === r.last && // either `null`, or a singular child
  (r.f & ne) === 0 && (r = r.first, (t & yt) !== 0 && (t & ke) !== 0 && r !== null && (r.f |= ke)), r !== null && (r.parent = i, i !== null && zi(r, i), w !== null && (w.f & k) !== 0 && (t & Ot) === 0)) {
    var o = (
      /** @type {Derived} */
      w
    );
    (o.effects ?? (o.effects = [])).push(r);
  }
  return s;
}
function vn() {
  return w !== null && !q;
}
function bn(t) {
  const e = gt(be, null, !1);
  return F(e, U), e.teardown = t, e;
}
function qi(t) {
  Ki();
  var e = (
    /** @type {Effect} */
    D.f
  ), n = !w && (e & mt) !== 0 && (e & Ie) === 0;
  if (n) {
    var i = (
      /** @type {ComponentContext} */
      Q
    );
    (i.e ?? (i.e = [])).push(t);
  } else
    return wn(t);
}
function wn(t) {
  return gt(qt | ci, t, !1);
}
function Se(t) {
  return gt(qt, t, !1);
}
function Xi(t) {
  return gt(Ee | ne, t, !0);
}
function Qi(t, e = 0) {
  return gt(be | e, t, !0);
}
function In(t, e = 0) {
  var n = gt(we | e, t, !0);
  return n;
}
function En(t) {
  return gt(mt | ne, t, !0);
}
function Nn(t) {
  var e = t.teardown;
  if (e !== null) {
    const n = rt, i = w;
    Fe(!0), Z(null);
    try {
      e.call(null);
    } finally {
      Fe(n), Z(i);
    }
  }
}
function Dn(t, e = !1) {
  var n = t.first;
  for (t.first = t.last = null; n !== null; ) {
    const s = n.ac;
    s !== null && Te(() => {
      s.abort(bt);
    });
    var i = n.next;
    (n.f & Ot) !== 0 ? n.parent = null : Ct(n, e), n = i;
  }
}
function Zi(t) {
  for (var e = t.first; e !== null; ) {
    var n = e.next;
    (e.f & mt) === 0 && Ct(e), e = n;
  }
}
function Ct(t, e = !0) {
  var n = !1;
  (e || (t.f & an) !== 0) && t.nodes !== null && t.nodes.end !== null && (Ji(
    t.nodes.start,
    /** @type {TemplateNode} */
    t.nodes.end
  ), n = !0), Dn(t, e && !n), Jt(t, 0), F(t, tt);
  var i = t.nodes && t.nodes.t;
  if (i !== null)
    for (const r of i)
      r.stop();
  Nn(t);
  var s = t.parent;
  s !== null && s.first !== null && Ln(t), t.next = t.prev = t.teardown = t.ctx = t.deps = t.fn = t.nodes = t.ac = null;
}
function Ji(t, e) {
  for (; t !== null; ) {
    var n = t === e ? null : /* @__PURE__ */ Yi(t);
    t.remove(), t = n;
  }
}
function Ln(t) {
  var e = t.parent, n = t.prev, i = t.next;
  n !== null && (n.next = i), i !== null && (i.prev = n), e !== null && (e.first === t && (e.first = i), e.last === t && (e.last = n));
}
let Kt = !1, rt = !1;
function Fe(t) {
  rt = t;
}
let w = null, q = !1;
function Z(t) {
  w = t;
}
let D = null;
function ot(t) {
  D = t;
}
let x = null;
function ts(t) {
  w !== null && (x === null ? x = [t] : x.push(t));
}
let H = null, B = 0, G = null;
function es(t) {
  G = t;
}
let An = 1, ct = 0, ht = ct;
function He(t) {
  ht = t;
}
function Tn() {
  return ++An;
}
function Bt(t) {
  var e = t.f;
  if ((e & j) !== 0)
    return !0;
  if (e & k && (t.f &= ~_t), (e & lt) !== 0) {
    for (var n = (
      /** @type {Value[]} */
      t.deps
    ), i = n.length, s = 0; s < i; s++) {
      var r = n[s];
      if (Bt(
        /** @type {Derived} */
        r
      ) && Cn(
        /** @type {Derived} */
        r
      ), r.wv > t.wv)
        return !0;
    }
    (e & $) !== 0 && // During time traveling we don't want to reset the status so that
    // traversal of the graph in the other batches still happens
    R === null && F(t, U);
  }
  return !1;
}
function Sn(t, e, n = !0) {
  var i = t.reactions;
  if (i !== null && !(x != null && x.includes(t)))
    for (var s = 0; s < i.length; s++) {
      var r = i[s];
      (r.f & k) !== 0 ? Sn(
        /** @type {Derived} */
        r,
        e,
        !1
      ) : e === r && (n ? F(r, j) : (r.f & U) !== 0 && F(r, lt), Pt(
        /** @type {Effect} */
        r
      ));
    }
}
function Pn(t) {
  var _;
  var e = H, n = B, i = G, s = w, r = x, o = Q, l = q, a = ht, u = t.f;
  H = /** @type {null | Value[]} */
  null, B = 0, G = null, w = (u & (mt | Ot)) === 0 ? t : null, x = null, Xt(t.ctx), q = !1, ht = ++ct, t.ac !== null && (Te(() => {
    t.ac.abort(bt);
  }), t.ac = null);
  try {
    t.f |= he;
    var c = (
      /** @type {Function} */
      t.fn
    ), d = c(), f = t.deps;
    if (H !== null) {
      var h;
      if (Jt(t, B), f !== null && B > 0)
        for (f.length = B + H.length, h = 0; h < H.length; h++)
          f[B + h] = H[h];
      else
        t.deps = f = H;
      if (vn() && (t.f & $) !== 0)
        for (h = B; h < f.length; h++)
          ((_ = f[h]).reactions ?? (_.reactions = [])).push(t);
    } else f !== null && B < f.length && (Jt(t, B), f.length = B);
    if (cn() && G !== null && !q && f !== null && (t.f & (k | lt | j)) === 0)
      for (h = 0; h < /** @type {Source[]} */
      G.length; h++)
        Sn(
          G[h],
          /** @type {Effect} */
          t
        );
    if (s !== null && s !== t) {
      if (ct++, s.deps !== null)
        for (let g = 0; g < n; g += 1)
          s.deps[g].rv = ct;
      if (e !== null)
        for (const g of e)
          g.rv = ct;
      G !== null && (i === null ? i = G : i.push(.../** @type {Source[]} */
      G));
    }
    return (t.f & it) !== 0 && (t.f ^= it), d;
  } catch (g) {
    return Ti(g);
  } finally {
    t.f ^= he, H = e, B = n, G = i, w = s, x = r, Xt(o), q = l, ht = a;
  }
}
function ns(t, e) {
  let n = e.reactions;
  if (n !== null) {
    var i = si.call(n, t);
    if (i !== -1) {
      var s = n.length - 1;
      s === 0 ? n = e.reactions = null : (n[i] = n[s], n.pop());
    }
  }
  if (n === null && (e.f & k) !== 0 && // Destroying a child effect while updating a parent effect can cause a dependency to appear
  // to be unused, when in fact it is used by the currently-updating parent. Checking `new_deps`
  // allows us to skip the expensive work of disconnecting and immediately reconnecting it
  (H === null || !H.includes(e))) {
    var r = (
      /** @type {Derived} */
      e
    );
    (r.f & $) !== 0 && (r.f ^= $, r.f &= ~_t), Ne(r), _n(r), Jt(r, 0);
  }
}
function Jt(t, e) {
  var n = t.deps;
  if (n !== null)
    for (var i = e; i < n.length; i++)
      ns(t, n[i]);
}
function Ft(t) {
  var e = t.f;
  if ((e & tt) === 0) {
    F(t, U);
    var n = D, i = Kt;
    D = t, Kt = !0;
    try {
      (e & (yt | we)) !== 0 ? Zi(t) : Dn(t), Nn(t);
      var s = Pn(t);
      t.teardown = typeof s == "function" ? s : null, t.wv = An;
      var r;
      fe && Ni && (t.f & j) !== 0 && t.deps;
    } finally {
      Kt = i, D = n;
    }
  }
}
function E(t) {
  var e = t.f, n = (e & k) !== 0;
  if (w !== null && !q) {
    var i = D !== null && (D.f & tt) !== 0;
    if (!i && !(x != null && x.includes(t))) {
      var s = w.deps;
      if ((w.f & he) !== 0)
        t.rv < ct && (t.rv = ct, H === null && s !== null && s[B] === t ? B++ : H === null ? H = [t] : H.push(t));
      else {
        (w.deps ?? (w.deps = [])).push(t);
        var r = t.reactions;
        r === null ? t.reactions = [w] : r.includes(w) || r.push(w);
      }
    }
  }
  if (rt && st.has(t))
    return st.get(t);
  if (n) {
    var o = (
      /** @type {Derived} */
      t
    );
    if (rt) {
      var l = o.v;
      return ((o.f & U) === 0 && o.reactions !== null || Rn(o)) && (l = Ae(o)), st.set(o, l), l;
    }
    var a = (o.f & $) === 0 && !q && w !== null && (Kt || (w.f & $) !== 0), u = o.deps === null;
    Bt(o) && (a && (o.f |= $), Cn(o)), a && !u && On(o);
  }
  if (R != null && R.has(t))
    return R.get(t);
  if ((t.f & it) !== 0)
    throw t.v;
  return t.v;
}
function On(t) {
  if (t.deps !== null) {
    t.f |= $;
    for (const e of t.deps)
      (e.reactions ?? (e.reactions = [])).push(t), (e.f & k) !== 0 && (e.f & $) === 0 && On(
        /** @type {Derived} */
        e
      );
  }
}
function Rn(t) {
  if (t.v === P) return !0;
  if (t.deps === null) return !1;
  for (const e of t.deps)
    if (st.has(e) || (e.f & k) !== 0 && Rn(
      /** @type {Derived} */
      e
    ))
      return !0;
  return !1;
}
function kn(t) {
  var e = q;
  try {
    return q = !0, t();
  } finally {
    q = e;
  }
}
const is = /* @__PURE__ */ new Set(), ss = /* @__PURE__ */ new Set();
function rs(t, e, n, i = {}) {
  function s(r) {
    if (i.capture || ls.call(e, r), !r.cancelBubble)
      return Te(() => n == null ? void 0 : n.call(this, r));
  }
  return t.startsWith("pointer") || t.startsWith("touch") || t === "wheel" ? xt(() => {
    e.addEventListener(t, s, i);
  }) : e.addEventListener(t, s, i), s;
}
function os(t) {
  for (var e = 0; e < t.length; e++)
    is.add(t[e]);
  for (var n of ss)
    n(t);
}
let Be = null;
function ls(t) {
  var b;
  var e = this, n = (
    /** @type {Node} */
    e.ownerDocument
  ), i = t.type, s = ((b = t.composedPath) == null ? void 0 : b.call(t)) || [], r = (
    /** @type {null | Element} */
    s[0] || t.target
  );
  Be = t;
  var o = 0, l = Be === t && t.__root;
  if (l) {
    var a = s.indexOf(l);
    if (a !== -1 && (e === document || e === /** @type {any} */
    window)) {
      t.__root = e;
      return;
    }
    var u = s.indexOf(e);
    if (u === -1)
      return;
    a <= u && (o = a);
  }
  if (r = /** @type {Element} */
  s[o] || t.target, r !== e) {
    ri(t, "currentTarget", {
      configurable: !0,
      get() {
        return r || n;
      }
    });
    var c = w, d = D;
    Z(null), ot(null);
    try {
      for (var f, h = []; r !== null; ) {
        var _ = r.assignedSlot || r.parentNode || /** @type {any} */
        r.host || null;
        try {
          var g = r["__" + i];
          g != null && (!/** @type {any} */
          r.disabled || // DOM could've been updated already by the time this is reached, so we check this as well
          // -> the target could not have been disabled because it emits the event in the first place
          t.target === r) && g.call(r, t);
        } catch (y) {
          f ? h.push(y) : f = y;
        }
        if (t.cancelBubble || _ === e || _ === null)
          break;
        r = _;
      }
      if (f) {
        for (let y of h)
          queueMicrotask(() => {
            throw y;
          });
        throw f;
      }
    } finally {
      t.__root = e, delete t.currentTarget, Z(c), ot(d);
    }
  }
}
function as(t) {
  var e = document.createElement("template");
  return e.innerHTML = t.replaceAll("<!>", "<!---->"), e.content;
}
function us(t, e) {
  var n = (
    /** @type {Effect} */
    D
  );
  n.nodes === null && (n.nodes = { start: t, end: e, a: null, t: null });
}
// @__NO_SIDE_EFFECTS__
function cs(t, e) {
  var n = (e & ei) !== 0, i, s = !t.startsWith("<!>");
  return () => {
    i === void 0 && (i = as(s ? t : "<!>" + t), i = /** @type {TemplateNode} */
    /* @__PURE__ */ Gi(i));
    var r = (
      /** @type {TemplateNode} */
      n || Vi ? document.importNode(i, !0) : i.cloneNode(!0)
    );
    return us(r, r), r;
  };
}
function ds(t, e) {
  t !== null && t.before(
    /** @type {Node} */
    e
  );
}
function fs(t) {
  return t.endsWith("capture") && t !== "gotpointercapture" && t !== "lostpointercapture";
}
const hs = [
  "beforeinput",
  "click",
  "change",
  "dblclick",
  "contextmenu",
  "focusin",
  "focusout",
  "input",
  "keydown",
  "keyup",
  "mousedown",
  "mousemove",
  "mouseout",
  "mouseover",
  "mouseup",
  "pointerdown",
  "pointermove",
  "pointerout",
  "pointerover",
  "pointerup",
  "touchend",
  "touchmove",
  "touchstart"
];
function ps(t) {
  return hs.includes(t);
}
const _s = {
  // no `class: 'className'` because we handle that separately
  formnovalidate: "formNoValidate",
  ismap: "isMap",
  nomodule: "noModule",
  playsinline: "playsInline",
  readonly: "readOnly",
  defaultvalue: "defaultValue",
  defaultchecked: "defaultChecked",
  srcobject: "srcObject",
  novalidate: "noValidate",
  allowfullscreen: "allowFullscreen",
  disablepictureinpicture: "disablePictureInPicture",
  disableremoteplayback: "disableRemotePlayback"
};
function Cs(t) {
  return t = t.toLowerCase(), _s[t] ?? t;
}
function ys(t, e) {
  var n = void 0, i;
  In(() => {
    n !== (n = e()) && (i && (Ct(i), i = null), n && (i = En(() => {
      Se(() => (
        /** @type {(node: Element) => void} */
        n(t)
      ));
    })));
  });
}
function Mn(t) {
  var e, n, i = "";
  if (typeof t == "string" || typeof t == "number") i += t;
  else if (typeof t == "object") if (Array.isArray(t)) {
    var s = t.length;
    for (e = 0; e < s; e++) t[e] && (n = Mn(t[e])) && (i && (i += " "), i += n);
  } else for (n in t) t[n] && (i && (i += " "), i += n);
  return i;
}
function ms() {
  for (var t, e, n = 0, i = "", s = arguments.length; n < s; n++) (t = arguments[n]) && (e = Mn(t)) && (i && (i += " "), i += e);
  return i;
}
function gs(t) {
  return typeof t == "object" ? ms(t) : t ?? "";
}
const Ve = [...` 	
\r\f \v\uFEFF`];
function vs(t, e, n) {
  var i = t == null ? "" : "" + t;
  if (n) {
    for (var s in n)
      if (n[s])
        i = i ? i + " " + s : s;
      else if (i.length)
        for (var r = s.length, o = 0; (o = i.indexOf(s, o)) >= 0; ) {
          var l = o + r;
          (o === 0 || Ve.includes(i[o - 1])) && (l === i.length || Ve.includes(i[l])) ? i = (o === 0 ? "" : i.substring(0, o)) + i.substring(l + 1) : o = l;
        }
  }
  return i === "" ? null : i;
}
function We(t, e = !1) {
  var n = e ? " !important;" : ";", i = "";
  for (var s in t) {
    var r = t[s];
    r != null && r !== "" && (i += " " + s + ": " + r + n);
  }
  return i;
}
function re(t) {
  return t[0] !== "-" || t[1] !== "-" ? t.toLowerCase() : t;
}
function bs(t, e) {
  if (e) {
    var n = "", i, s;
    if (Array.isArray(e) ? (i = e[0], s = e[1]) : i = e, t) {
      t = String(t).replaceAll(/\s*\/\*.*?\*\/\s*/g, "").trim();
      var r = !1, o = 0, l = !1, a = [];
      i && a.push(...Object.keys(i).map(re)), s && a.push(...Object.keys(s).map(re));
      var u = 0, c = -1;
      const g = t.length;
      for (var d = 0; d < g; d++) {
        var f = t[d];
        if (l ? f === "/" && t[d - 1] === "*" && (l = !1) : r ? r === f && (r = !1) : f === "/" && t[d + 1] === "*" ? l = !0 : f === '"' || f === "'" ? r = f : f === "(" ? o++ : f === ")" && o--, !l && r === !1 && o === 0) {
          if (f === ":" && c === -1)
            c = d;
          else if (f === ";" || d === g - 1) {
            if (c !== -1) {
              var h = re(t.substring(u, c).trim());
              if (!a.includes(h)) {
                f !== ";" && d++;
                var _ = t.substring(u, d).trim();
                n += " " + _ + ";";
              }
            }
            u = d + 1, c = -1;
          }
        }
      }
    }
    return i && (n += We(i)), s && (n += We(s, !0)), n = n.trim(), n === "" ? null : n;
  }
  return t == null ? null : String(t);
}
function ws(t, e, n, i, s, r) {
  var o = t.__className;
  if (o !== n || o === void 0) {
    var l = vs(n, i, r);
    l == null ? t.removeAttribute("class") : e ? t.className = l : t.setAttribute("class", l), t.__className = n;
  } else if (r && s !== r)
    for (var a in r) {
      var u = !!r[a];
      (s == null || u !== !!s[a]) && t.classList.toggle(a, u);
    }
  return r;
}
function oe(t, e = {}, n, i) {
  for (var s in n) {
    var r = n[s];
    e[s] !== r && (n[s] == null ? t.style.removeProperty(s) : t.style.setProperty(s, r, i));
  }
}
function Is(t, e, n, i) {
  var s = t.__style;
  if (s !== e) {
    var r = bs(e, i);
    r == null ? t.removeAttribute("style") : t.style.cssText = r, t.__style = e;
  } else i && (Array.isArray(i) ? (oe(t, n == null ? void 0 : n[0], i[0]), oe(t, n == null ? void 0 : n[1], i[1], "important")) : oe(t, n, i));
  return i;
}
function ve(t, e, n = !1) {
  if (t.multiple) {
    if (e == null)
      return;
    if (!sn(e))
      return wi();
    for (var i of t.options)
      i.selected = e.includes(je(i));
    return;
  }
  for (i of t.options) {
    var s = je(i);
    if (Bi(s, e)) {
      i.selected = !0;
      return;
    }
  }
  (!n || e !== void 0) && (t.selectedIndex = -1);
}
function Es(t) {
  var e = new MutationObserver(() => {
    ve(t, t.__value);
  });
  e.observe(t, {
    // Listen to option element changes
    childList: !0,
    subtree: !0,
    // because of <optgroup>
    // Listen to option element value attribute changes
    // (doesn't get notified of select value changes,
    // because that property is not reflected as an attribute)
    attributes: !0,
    attributeFilter: ["value"]
  }), bn(() => {
    e.disconnect();
  });
}
function je(t) {
  return "__value" in t ? t.__value : t.value;
}
const Rt = Symbol("class"), kt = Symbol("style"), Un = Symbol("is custom element"), xn = Symbol("is html");
function Ns(t, e) {
  e ? t.hasAttribute("selected") || t.setAttribute("selected", "") : t.removeAttribute("selected");
}
function Ge(t, e, n, i) {
  var s = Fn(t);
  s[e] !== (s[e] = n) && (e === "loading" && (t[fi] = n), n == null ? t.removeAttribute(e) : typeof n != "string" && Hn(t).includes(e) ? t[e] = n : t.setAttribute(e, n));
}
function Ds(t, e, n, i, s = !1, r = !1) {
  var o = Fn(t), l = o[Un], a = !o[xn], u = e || {}, c = t.tagName === "OPTION";
  for (var d in e)
    d in n || (n[d] = null);
  n.class ? n.class = gs(n.class) : n[Rt] && (n.class = null), n[kt] && (n.style ?? (n.style = null));
  var f = Hn(t);
  for (const p in n) {
    let v = n[p];
    if (c && p === "value" && v == null) {
      t.value = t.__value = "", u[p] = v;
      continue;
    }
    if (p === "class") {
      var h = t.namespaceURI === "http://www.w3.org/1999/xhtml";
      ws(t, h, v, i, e == null ? void 0 : e[Rt], n[Rt]), u[p] = v, u[Rt] = n[Rt];
      continue;
    }
    if (p === "style") {
      Is(t, v, e == null ? void 0 : e[kt], n[kt]), u[p] = v, u[kt] = n[kt];
      continue;
    }
    var _ = u[p];
    if (!(v === _ && !(v === void 0 && t.hasAttribute(p)))) {
      u[p] = v;
      var g = p[0] + p[1];
      if (g !== "$$")
        if (g === "on") {
          const S = {}, at = "$$" + p;
          let O = p.slice(2);
          var b = ps(O);
          if (fs(O) && (O = O.slice(0, -7), S.capture = !0), !b && _) {
            if (v != null) continue;
            t.removeEventListener(O, u[at], S), u[at] = null;
          }
          if (v != null)
            if (b)
              t[`__${O}`] = v, os([O]);
            else {
              let Gn = function(Yn) {
                u[p].call(this, Yn);
              };
              u[at] = rs(O, t, Gn, S);
            }
          else b && (t[`__${O}`] = void 0);
        } else if (p === "style")
          Ge(t, p, v);
        else if (p === "autofocus")
          $i(
            /** @type {HTMLElement} */
            t,
            !!v
          );
        else if (!l && (p === "__value" || p === "value" && v != null))
          t.value = t.__value = v;
        else if (p === "selected" && c)
          Ns(
            /** @type {HTMLOptionElement} */
            t,
            v
          );
        else {
          var y = p;
          a || (y = Cs(y));
          var m = y === "defaultValue" || y === "defaultChecked";
          if (v == null && !l && !m)
            if (o[p] = null, y === "value" || y === "checked") {
              let S = (
                /** @type {HTMLInputElement} */
                t
              );
              const at = e === void 0;
              if (y === "value") {
                let O = S.defaultValue;
                S.removeAttribute(y), S.defaultValue = O, S.value = S.__value = at ? O : null;
              } else {
                let O = S.defaultChecked;
                S.removeAttribute(y), S.defaultChecked = O, S.checked = at ? O : !1;
              }
            } else
              t.removeAttribute(p);
          else m || f.includes(y) && (l || typeof v != "string") ? (t[y] = v, y in o && (o[y] = P)) : typeof v != "function" && Ge(t, y, v);
        }
    }
  }
  return u;
}
function Ls(t, e, n = [], i = [], s = [], r, o = !1, l = !1) {
  ki(s, n, i, (a) => {
    var u = void 0, c = {}, d = t.nodeName === "SELECT", f = !1;
    if (In(() => {
      var _ = e(...a.map(E)), g = Ds(
        t,
        u,
        _,
        r,
        o,
        l
      );
      f && d && "value" in _ && ve(
        /** @type {HTMLSelectElement} */
        t,
        _.value
      );
      for (let y of Object.getOwnPropertySymbols(c))
        _[y] || Ct(c[y]);
      for (let y of Object.getOwnPropertySymbols(_)) {
        var b = _[y];
        y.description === ii && (!u || b !== u[y]) && (c[y] && Ct(c[y]), c[y] = En(() => ys(t, () => b))), g[y] = b;
      }
      u = g;
    }), d) {
      var h = (
        /** @type {HTMLSelectElement} */
        t
      );
      Se(() => {
        ve(
          h,
          /** @type {Record<string | symbol, any>} */
          u.value,
          !0
        ), Es(h);
      });
    }
    f = !0;
  });
}
function Fn(t) {
  return (
    /** @type {Record<string | symbol, unknown>} **/
    // @ts-expect-error
    t.__attributes ?? (t.__attributes = {
      [Un]: t.nodeName.includes("-"),
      [xn]: t.namespaceURI === ni
    })
  );
}
var Ye = /* @__PURE__ */ new Map();
function Hn(t) {
  var e = t.getAttribute("is") || t.nodeName, n = Ye.get(e);
  if (n) return n;
  Ye.set(e, n = []);
  for (var i, s = t, r = Element.prototype; r !== s; ) {
    i = oi(s);
    for (var o in i)
      i[o].set && n.push(o);
    s = rn(s);
  }
  return n;
}
function $e(t, e) {
  return t === e || (t == null ? void 0 : t[ft]) === e;
}
function As(t = {}, e, n, i) {
  return Se(() => {
    var s, r;
    return Qi(() => {
      s = r, r = [], kn(() => {
        t !== n(...r) && (e(t, ...r), s && $e(n(...s), t) && e(null, ...s));
      });
    }), () => {
      xt(() => {
        r && $e(n(...r), t) && e(null, ...r);
      });
    };
  }), t;
}
let Wt = !1;
function Ts(t) {
  var e = Wt;
  try {
    return Wt = !1, [t(), Wt];
  } finally {
    Wt = e;
  }
}
function jt(t, e, n, i) {
  var m;
  var s = (n & Jn) !== 0, r = (n & ti) !== 0, o = (
    /** @type {V} */
    i
  ), l = !0, a = () => (l && (l = !1, o = r ? kn(
    /** @type {() => V} */
    i
  ) : (
    /** @type {V} */
    i
  )), o), u;
  if (s) {
    var c = ft in t || di in t;
    u = ((m = $t(t, e)) == null ? void 0 : m.set) ?? (c && e in t ? (p) => t[e] = p : void 0);
  }
  var d, f = !1;
  s ? [d, f] = Ts(() => (
    /** @type {V} */
    t[e]
  )) : d = /** @type {V} */
  t[e], d === void 0 && i !== void 0 && (d = a(), u && (mi(), u(d)));
  var h;
  if (h = () => {
    var p = (
      /** @type {V} */
      t[e]
    );
    return p === void 0 ? a() : (l = !0, p);
  }, (n & Zn) === 0)
    return h;
  if (u) {
    var _ = t.$$legacy;
    return (
      /** @type {() => V} */
      (function(p, v) {
        return arguments.length > 0 ? ((!v || _ || f) && u(v ? h() : p), p) : h();
      })
    );
  }
  var g = !1, b = ((n & Qn) !== 0 ? Le : xi)(() => (g = !1, h()));
  s && E(b);
  var y = (
    /** @type {Effect} */
    D
  );
  return (
    /** @type {() => V} */
    (function(p, v) {
      if (arguments.length > 0) {
        const S = v ? E(b) : s ? It(p) : p;
        return W(b, S), g = !0, o !== void 0 && (o = S), p;
      }
      return rt && g || (y.f & tt) !== 0 ? b.v : E(b);
    })
  );
}
const Ss = [
  [
    "af",
    // Afghanistan
    "93",
    0,
    null,
    "0"
  ],
  [
    "ax",
    // Åland Islands
    "358",
    1,
    ["18", "4"],
    // (4 is a mobile range shared with FI)
    "0"
  ],
  [
    "al",
    // Albania
    "355",
    0,
    null,
    "0"
  ],
  [
    "dz",
    // Algeria
    "213",
    0,
    null,
    "0"
  ],
  [
    "as",
    // American Samoa
    "1",
    5,
    ["684"],
    "1"
  ],
  [
    "ad",
    // Andorra
    "376"
  ],
  [
    "ao",
    // Angola
    "244"
  ],
  [
    "ai",
    // Anguilla
    "1",
    6,
    ["264"],
    "1"
  ],
  [
    "ag",
    // Antigua and Barbuda
    "1",
    7,
    ["268"],
    "1"
  ],
  [
    "ar",
    // Argentina
    "54",
    0,
    null,
    "0"
  ],
  [
    "am",
    // Armenia
    "374",
    0,
    null,
    "0"
  ],
  [
    "aw",
    // Aruba
    "297"
  ],
  [
    "ac",
    // Ascension Island
    "247"
  ],
  [
    "au",
    // Australia
    "61",
    0,
    ["4"],
    // (mobile range shared with CX and CC)
    "0"
  ],
  [
    "at",
    // Austria
    "43",
    0,
    null,
    "0"
  ],
  [
    "az",
    // Azerbaijan
    "994",
    0,
    null,
    "0"
  ],
  [
    "bs",
    // Bahamas
    "1",
    8,
    ["242"],
    "1"
  ],
  [
    "bh",
    // Bahrain
    "973"
  ],
  [
    "bd",
    // Bangladesh
    "880",
    0,
    null,
    "0"
  ],
  [
    "bb",
    // Barbados
    "1",
    9,
    ["246"],
    "1"
  ],
  [
    "by",
    // Belarus
    "375",
    0,
    null,
    "8"
  ],
  [
    "be",
    // Belgium
    "32",
    0,
    null,
    "0"
  ],
  [
    "bz",
    // Belize
    "501"
  ],
  [
    "bj",
    // Benin
    "229"
  ],
  [
    "bm",
    // Bermuda
    "1",
    10,
    ["441"],
    "1"
  ],
  [
    "bt",
    // Bhutan
    "975"
  ],
  [
    "bo",
    // Bolivia
    "591",
    0,
    null,
    "0"
  ],
  [
    "ba",
    // Bosnia and Herzegovina
    "387",
    0,
    null,
    "0"
  ],
  [
    "bw",
    // Botswana
    "267"
  ],
  [
    "br",
    // Brazil
    "55",
    0,
    null,
    "0"
  ],
  [
    "io",
    // British Indian Ocean Territory
    "246"
  ],
  [
    "vg",
    // British Virgin Islands
    "1",
    11,
    ["284"],
    "1"
  ],
  [
    "bn",
    // Brunei
    "673"
  ],
  [
    "bg",
    // Bulgaria
    "359",
    0,
    null,
    "0"
  ],
  [
    "bf",
    // Burkina Faso
    "226"
  ],
  [
    "bi",
    // Burundi
    "257"
  ],
  [
    "kh",
    // Cambodia
    "855",
    0,
    null,
    "0"
  ],
  [
    "cm",
    // Cameroon
    "237"
  ],
  [
    "ca",
    // Canada
    "1",
    1,
    ["204", "226", "236", "249", "250", "257", "263", "289", "306", "343", "354", "365", "367", "368", "382", "403", "416", "418", "428", "431", "437", "438", "450", "468", "474", "506", "514", "519", "548", "579", "581", "584", "587", "604", "613", "639", "647", "672", "683", "705", "709", "742", "753", "778", "780", "782", "807", "819", "825", "867", "873", "879", "902", "905", "942"],
    "1"
  ],
  [
    "cv",
    // Cape Verde
    "238"
  ],
  [
    "bq",
    // Caribbean Netherlands
    "599",
    1,
    ["3", "4", "7"]
  ],
  [
    "ky",
    // Cayman Islands
    "1",
    12,
    ["345"],
    "1"
  ],
  [
    "cf",
    // Central African Republic
    "236"
  ],
  [
    "td",
    // Chad
    "235"
  ],
  [
    "cl",
    // Chile
    "56"
  ],
  [
    "cn",
    // China
    "86",
    0,
    null,
    "0"
  ],
  [
    "cx",
    // Christmas Island
    "61",
    2,
    ["4", "89164"],
    // (4 is a mobile range shared with AU and CC)
    "0"
  ],
  [
    "cc",
    // Cocos (Keeling) Islands
    "61",
    1,
    ["4", "89162"],
    // (4 is a mobile range shared with AU and CX)
    "0"
  ],
  [
    "co",
    // Colombia
    "57",
    0,
    null,
    "0"
  ],
  [
    "km",
    // Comoros
    "269"
  ],
  [
    "cg",
    // Congo (Brazzaville)
    "242"
  ],
  [
    "cd",
    // Congo (Kinshasa)
    "243",
    0,
    null,
    "0"
  ],
  [
    "ck",
    // Cook Islands
    "682"
  ],
  [
    "cr",
    // Costa Rica
    "506"
  ],
  [
    "ci",
    // Côte d'Ivoire
    "225"
  ],
  [
    "hr",
    // Croatia
    "385",
    0,
    null,
    "0"
  ],
  [
    "cu",
    // Cuba
    "53",
    0,
    null,
    "0"
  ],
  [
    "cw",
    // Curaçao
    "599",
    0
  ],
  [
    "cy",
    // Cyprus
    "357"
  ],
  [
    "cz",
    // Czech Republic
    "420"
  ],
  [
    "dk",
    // Denmark
    "45"
  ],
  [
    "dj",
    // Djibouti
    "253"
  ],
  [
    "dm",
    // Dominica
    "1",
    13,
    ["767"],
    "1"
  ],
  [
    "do",
    // Dominican Republic
    "1",
    2,
    ["809", "829", "849"],
    "1"
  ],
  [
    "ec",
    // Ecuador
    "593",
    0,
    null,
    "0"
  ],
  [
    "eg",
    // Egypt
    "20",
    0,
    null,
    "0"
  ],
  [
    "sv",
    // El Salvador
    "503"
  ],
  [
    "gq",
    // Equatorial Guinea
    "240"
  ],
  [
    "er",
    // Eritrea
    "291",
    0,
    null,
    "0"
  ],
  [
    "ee",
    // Estonia
    "372"
  ],
  [
    "sz",
    // Eswatini
    "268"
  ],
  [
    "et",
    // Ethiopia
    "251",
    0,
    null,
    "0"
  ],
  [
    "fk",
    // Falkland Islands (Malvinas)
    "500"
  ],
  [
    "fo",
    // Faroe Islands
    "298"
  ],
  [
    "fj",
    // Fiji
    "679"
  ],
  [
    "fi",
    // Finland
    "358",
    0,
    ["4"],
    // (mobile range shared with AX)
    "0"
  ],
  [
    "fr",
    // France
    "33",
    0,
    null,
    "0"
  ],
  [
    "gf",
    // French Guiana
    "594",
    0,
    null,
    "0"
  ],
  [
    "pf",
    // French Polynesia
    "689"
  ],
  [
    "ga",
    // Gabon
    "241"
  ],
  [
    "gm",
    // Gambia
    "220"
  ],
  [
    "ge",
    // Georgia
    "995",
    0,
    null,
    "0"
  ],
  [
    "de",
    // Germany
    "49",
    0,
    null,
    "0"
  ],
  [
    "gh",
    // Ghana
    "233",
    0,
    null,
    "0"
  ],
  [
    "gi",
    // Gibraltar
    "350"
  ],
  [
    "gr",
    // Greece
    "30"
  ],
  [
    "gl",
    // Greenland
    "299"
  ],
  [
    "gd",
    // Grenada
    "1",
    14,
    ["473"],
    "1"
  ],
  [
    "gp",
    // Guadeloupe
    "590",
    0,
    null,
    "0"
  ],
  [
    "gu",
    // Guam
    "1",
    15,
    ["671"],
    "1"
  ],
  [
    "gt",
    // Guatemala
    "502"
  ],
  [
    "gg",
    // Guernsey
    "44",
    1,
    ["1481", "7781", "7839", "7911"],
    "0"
  ],
  [
    "gn",
    // Guinea
    "224"
  ],
  [
    "gw",
    // Guinea-Bissau
    "245"
  ],
  [
    "gy",
    // Guyana
    "592"
  ],
  [
    "ht",
    // Haiti
    "509"
  ],
  [
    "hn",
    // Honduras
    "504"
  ],
  [
    "hk",
    // Hong Kong SAR China
    "852"
  ],
  [
    "hu",
    // Hungary
    "36",
    0,
    null,
    "06"
  ],
  [
    "is",
    // Iceland
    "354"
  ],
  [
    "in",
    // India
    "91",
    0,
    null,
    "0"
  ],
  [
    "id",
    // Indonesia
    "62",
    0,
    null,
    "0"
  ],
  [
    "ir",
    // Iran
    "98",
    0,
    null,
    "0"
  ],
  [
    "iq",
    // Iraq
    "964",
    0,
    null,
    "0"
  ],
  [
    "ie",
    // Ireland
    "353",
    0,
    null,
    "0"
  ],
  [
    "im",
    // Isle of Man
    "44",
    2,
    ["1624", "74576", "7524", "7624", "7924"],
    "0"
  ],
  [
    "il",
    // Israel
    "972",
    0,
    null,
    "0"
  ],
  [
    "it",
    // Italy
    "39",
    0,
    ["3"]
    // (mobile range shared with VA)
  ],
  [
    "jm",
    // Jamaica
    "1",
    4,
    ["658", "876"],
    "1"
  ],
  [
    "jp",
    // Japan
    "81",
    0,
    null,
    "0"
  ],
  [
    "je",
    // Jersey
    "44",
    3,
    ["1534", "7509", "7700", "7797", "7829", "7937"],
    "0"
  ],
  [
    "jo",
    // Jordan
    "962",
    0,
    null,
    "0"
  ],
  [
    "kz",
    // Kazakhstan
    "7",
    1,
    ["33", "7"],
    // (33 is shared with RU)
    "8"
  ],
  [
    "ke",
    // Kenya
    "254",
    0,
    null,
    "0"
  ],
  [
    "ki",
    // Kiribati
    "686",
    0,
    null,
    "0"
  ],
  [
    "xk",
    // Kosovo
    "383",
    0,
    null,
    "0"
  ],
  [
    "kw",
    // Kuwait
    "965"
  ],
  [
    "kg",
    // Kyrgyzstan
    "996",
    0,
    null,
    "0"
  ],
  [
    "la",
    // Laos
    "856",
    0,
    null,
    "0"
  ],
  [
    "lv",
    // Latvia
    "371"
  ],
  [
    "lb",
    // Lebanon
    "961",
    0,
    null,
    "0"
  ],
  [
    "ls",
    // Lesotho
    "266"
  ],
  [
    "lr",
    // Liberia
    "231",
    0,
    null,
    "0"
  ],
  [
    "ly",
    // Libya
    "218",
    0,
    null,
    "0"
  ],
  [
    "li",
    // Liechtenstein
    "423",
    0,
    null,
    "0"
  ],
  [
    "lt",
    // Lithuania
    "370",
    0,
    null,
    "0"
  ],
  [
    "lu",
    // Luxembourg
    "352"
  ],
  [
    "mo",
    // Macao SAR China
    "853"
  ],
  [
    "mg",
    // Madagascar
    "261",
    0,
    null,
    "0"
  ],
  [
    "mw",
    // Malawi
    "265",
    0,
    null,
    "0"
  ],
  [
    "my",
    // Malaysia
    "60",
    0,
    null,
    "0"
  ],
  [
    "mv",
    // Maldives
    "960"
  ],
  [
    "ml",
    // Mali
    "223"
  ],
  [
    "mt",
    // Malta
    "356"
  ],
  [
    "mh",
    // Marshall Islands
    "692",
    0,
    null,
    "1"
  ],
  [
    "mq",
    // Martinique
    "596",
    0,
    null,
    "0"
  ],
  [
    "mr",
    // Mauritania
    "222"
  ],
  [
    "mu",
    // Mauritius
    "230"
  ],
  [
    "yt",
    // Mayotte
    "262",
    1,
    ["269", "639"],
    "0"
  ],
  [
    "mx",
    // Mexico
    "52"
  ],
  [
    "fm",
    // Micronesia
    "691"
  ],
  [
    "md",
    // Moldova
    "373",
    0,
    null,
    "0"
  ],
  [
    "mc",
    // Monaco
    "377",
    0,
    null,
    "0"
  ],
  [
    "mn",
    // Mongolia
    "976",
    0,
    null,
    "0"
  ],
  [
    "me",
    // Montenegro
    "382",
    0,
    null,
    "0"
  ],
  [
    "ms",
    // Montserrat
    "1",
    16,
    ["664"],
    "1"
  ],
  [
    "ma",
    // Morocco
    "212",
    0,
    ["6", "7"],
    // (mobile ranges shared with EH)
    "0"
  ],
  [
    "mz",
    // Mozambique
    "258"
  ],
  [
    "mm",
    // Myanmar (Burma)
    "95",
    0,
    null,
    "0"
  ],
  [
    "na",
    // Namibia
    "264",
    0,
    null,
    "0"
  ],
  [
    "nr",
    // Nauru
    "674"
  ],
  [
    "np",
    // Nepal
    "977",
    0,
    null,
    "0"
  ],
  [
    "nl",
    // Netherlands
    "31",
    0,
    null,
    "0"
  ],
  [
    "nc",
    // New Caledonia
    "687"
  ],
  [
    "nz",
    // New Zealand
    "64",
    0,
    null,
    "0"
  ],
  [
    "ni",
    // Nicaragua
    "505"
  ],
  [
    "ne",
    // Niger
    "227"
  ],
  [
    "ng",
    // Nigeria
    "234",
    0,
    null,
    "0"
  ],
  [
    "nu",
    // Niue
    "683"
  ],
  [
    "nf",
    // Norfolk Island
    "672"
  ],
  [
    "kp",
    // North Korea
    "850",
    0,
    null,
    "0"
  ],
  [
    "mk",
    // North Macedonia
    "389",
    0,
    null,
    "0"
  ],
  [
    "mp",
    // Northern Mariana Islands
    "1",
    17,
    ["670"],
    "1"
  ],
  [
    "no",
    // Norway
    "47",
    0,
    ["4", "9"]
    // (mobile ranges shared with SJ)
  ],
  [
    "om",
    // Oman
    "968"
  ],
  [
    "pk",
    // Pakistan
    "92",
    0,
    null,
    "0"
  ],
  [
    "pw",
    // Palau
    "680"
  ],
  [
    "ps",
    // Palestinian Territories
    "970",
    0,
    null,
    "0"
  ],
  [
    "pa",
    // Panama
    "507"
  ],
  [
    "pg",
    // Papua New Guinea
    "675"
  ],
  [
    "py",
    // Paraguay
    "595",
    0,
    null,
    "0"
  ],
  [
    "pe",
    // Peru
    "51",
    0,
    null,
    "0"
  ],
  [
    "ph",
    // Philippines
    "63",
    0,
    null,
    "0"
  ],
  [
    "pl",
    // Poland
    "48"
  ],
  [
    "pt",
    // Portugal
    "351"
  ],
  [
    "pr",
    // Puerto Rico
    "1",
    3,
    ["787", "939"],
    "1"
  ],
  [
    "qa",
    // Qatar
    "974"
  ],
  [
    "re",
    // Réunion
    "262",
    0,
    null,
    "0"
  ],
  [
    "ro",
    // Romania
    "40",
    0,
    null,
    "0"
  ],
  [
    "ru",
    // Russia
    "7",
    0,
    ["33"],
    // (shared with KZ)
    "8"
  ],
  [
    "rw",
    // Rwanda
    "250",
    0,
    null,
    "0"
  ],
  [
    "ws",
    // Samoa
    "685"
  ],
  [
    "sm",
    // San Marino
    "378"
  ],
  [
    "st",
    // São Tomé & Príncipe
    "239"
  ],
  [
    "sa",
    // Saudi Arabia
    "966",
    0,
    null,
    "0"
  ],
  [
    "sn",
    // Senegal
    "221"
  ],
  [
    "rs",
    // Serbia
    "381",
    0,
    null,
    "0"
  ],
  [
    "sc",
    // Seychelles
    "248"
  ],
  [
    "sl",
    // Sierra Leone
    "232",
    0,
    null,
    "0"
  ],
  [
    "sg",
    // Singapore
    "65"
  ],
  [
    "sx",
    // Sint Maarten
    "1",
    21,
    ["721"],
    "1"
  ],
  [
    "sk",
    // Slovakia
    "421",
    0,
    null,
    "0"
  ],
  [
    "si",
    // Slovenia
    "386",
    0,
    null,
    "0"
  ],
  [
    "sb",
    // Solomon Islands
    "677"
  ],
  [
    "so",
    // Somalia
    "252",
    0,
    null,
    "0"
  ],
  [
    "za",
    // South Africa
    "27",
    0,
    null,
    "0"
  ],
  [
    "kr",
    // South Korea
    "82",
    0,
    null,
    "0"
  ],
  [
    "ss",
    // South Sudan
    "211",
    0,
    null,
    "0"
  ],
  [
    "es",
    // Spain
    "34"
  ],
  [
    "lk",
    // Sri Lanka
    "94",
    0,
    null,
    "0"
  ],
  [
    "bl",
    // St. Barthélemy
    "590",
    1,
    null,
    "0"
  ],
  [
    "sh",
    // St. Helena
    "290"
  ],
  [
    "kn",
    // St. Kitts & Nevis
    "1",
    18,
    ["869"],
    "1"
  ],
  [
    "lc",
    // St. Lucia
    "1",
    19,
    ["758"],
    "1"
  ],
  [
    "mf",
    // St. Martin
    "590",
    2,
    null,
    "0"
  ],
  [
    "pm",
    // St. Pierre & Miquelon
    "508",
    0,
    null,
    "0"
  ],
  [
    "vc",
    // St. Vincent & Grenadines
    "1",
    20,
    ["784"],
    "1"
  ],
  [
    "sd",
    // Sudan
    "249",
    0,
    null,
    "0"
  ],
  [
    "sr",
    // Suriname
    "597"
  ],
  [
    "sj",
    // Svalbard & Jan Mayen
    "47",
    1,
    ["4", "79", "9"]
    // (4 and 9 are mobile ranges shared with NO)
  ],
  [
    "se",
    // Sweden
    "46",
    0,
    null,
    "0"
  ],
  [
    "ch",
    // Switzerland
    "41",
    0,
    null,
    "0"
  ],
  [
    "sy",
    // Syria
    "963",
    0,
    null,
    "0"
  ],
  [
    "tw",
    // Taiwan
    "886",
    0,
    null,
    "0"
  ],
  [
    "tj",
    // Tajikistan
    "992"
  ],
  [
    "tz",
    // Tanzania
    "255",
    0,
    null,
    "0"
  ],
  [
    "th",
    // Thailand
    "66",
    0,
    null,
    "0"
  ],
  [
    "tl",
    // Timor-Leste
    "670"
  ],
  [
    "tg",
    // Togo
    "228"
  ],
  [
    "tk",
    // Tokelau
    "690"
  ],
  [
    "to",
    // Tonga
    "676"
  ],
  [
    "tt",
    // Trinidad & Tobago
    "1",
    22,
    ["868"],
    "1"
  ],
  [
    "tn",
    // Tunisia
    "216"
  ],
  [
    "tr",
    // Turkey
    "90",
    0,
    null,
    "0"
  ],
  [
    "tm",
    // Turkmenistan
    "993",
    0,
    null,
    "8"
  ],
  [
    "tc",
    // Turks & Caicos Islands
    "1",
    23,
    ["649"],
    "1"
  ],
  [
    "tv",
    // Tuvalu
    "688"
  ],
  [
    "vi",
    // U.S. Virgin Islands
    "1",
    24,
    ["340"],
    "1"
  ],
  [
    "ug",
    // Uganda
    "256",
    0,
    null,
    "0"
  ],
  [
    "ua",
    // Ukraine
    "380",
    0,
    null,
    "0"
  ],
  [
    "ae",
    // United Arab Emirates
    "971",
    0,
    null,
    "0"
  ],
  [
    "gb",
    // United Kingdom
    "44",
    0,
    null,
    "0"
  ],
  [
    "us",
    // United States
    "1",
    0,
    null,
    "1"
  ],
  [
    "uy",
    // Uruguay
    "598",
    0,
    null,
    "0"
  ],
  [
    "uz",
    // Uzbekistan
    "998"
  ],
  [
    "vu",
    // Vanuatu
    "678"
  ],
  [
    "va",
    // Vatican City
    "39",
    1,
    ["06698", "3"]
    // (3 is a mobile range shared with IT)
  ],
  [
    "ve",
    // Venezuela
    "58",
    0,
    null,
    "0"
  ],
  [
    "vn",
    // Vietnam
    "84",
    0,
    null,
    "0"
  ],
  [
    "wf",
    // Wallis & Futuna
    "681"
  ],
  [
    "eh",
    // Western Sahara
    "212",
    1,
    ["5288", "5289", "6", "7"],
    // (6 and 7 are mobile ranges shared with MA)
    "0"
  ],
  [
    "ye",
    // Yemen
    "967",
    0,
    null,
    "0"
  ],
  [
    "zm",
    // Zambia
    "260",
    0,
    null,
    "0"
  ],
  [
    "zw",
    // Zimbabwe
    "263",
    0,
    null,
    "0"
  ]
], pt = [];
for (const t of Ss)
  pt.push({
    name: "",
    // populated in the plugin
    iso2: t[0],
    dialCode: t[1],
    priority: t[2] || 0,
    areaCodes: t[3] || null,
    nodeById: {},
    // populated by the plugin
    nationalPrefix: t[4] || null,
    normalisedName: "",
    // populated in the plugin
    initials: "",
    // populated in the plugin
    dialCodePlus: ""
    // populated in the plugin
  });
const Gt = {
  OPEN_COUNTRY_DROPDOWN: "open:countrydropdown",
  CLOSE_COUNTRY_DROPDOWN: "close:countrydropdown",
  COUNTRY_CHANGE: "countrychange",
  INPUT: "input"
  // used for synthetic input trigger
}, I = {
  HIDE: "iti__hide",
  V_HIDE: "iti__v-hide",
  ARROW_UP: "iti__arrow--up",
  GLOBE: "iti__globe",
  FLAG: "iti__flag",
  LOADING: "iti__loading",
  COUNTRY_ITEM: "iti__country",
  HIGHLIGHT: "iti__highlight"
}, M = {
  ARROW_UP: "ArrowUp",
  ARROW_DOWN: "ArrowDown",
  SPACE: " ",
  ENTER: "Enter",
  ESC: "Escape",
  TAB: "Tab"
}, Ke = {
  PASTE: "insertFromPaste",
  DELETE_FWD: "deleteContentForward"
}, ut = {
  ALPHA_UNICODE: new RegExp("\\p{L}", "u"),
  // any kind of letter from any language
  NON_PLUS_NUMERIC: /[^+0-9]/,
  // chars that are NOT + or digit
  NON_PLUS_NUMERIC_GLOBAL: /[^+0-9]/g,
  // chars that are NOT + or digit (global)
  HIDDEN_SEARCH_CHAR: /^[a-zA-ZÀ-ÿа-яА-Я ]$/
  // single acceptable hidden-search char
}, Ps = {
  HIDDEN_SEARCH_RESET_MS: 1e3
}, ze = {
  UNKNOWN_NUMBER_TYPE: -99,
  UNKNOWN_VALIDATION_ERROR: -99
}, le = {
  SANE_SELECTED_WITH_DIAL_WIDTH: 78,
  // px width fallback when separateDialCode enabled
  SANE_SELECTED_NO_DIAL_WIDTH: 42,
  // px width fallback when no separate dial code
  INPUT_PADDING_EXTRA_LEFT: 6
  // px gap between selected country container and input text
}, Bn = {
  NANP: "1"
  // North American Numbering Plan
}, ae = {
  DIAL_CODE: "44",
  // +44 United Kingdom
  MOBILE_PREFIX: "7",
  // UK mobile numbers start with 7 after national trunk (0) or core section
  MOBILE_CORE_LENGTH: 10
  // core number length (excluding dial code / national prefix) for mobiles
}, Os = {
  ISO2: "us"
}, te = {
  AGGRESSIVE: "aggressive",
  POLITE: "polite",
  OFF: "off"
}, Mt = {
  AUTO: "auto"
}, Rs = [
  "FIXED_LINE",
  "MOBILE",
  "FIXED_LINE_OR_MOBILE",
  "TOLL_FREE",
  "PREMIUM_RATE",
  "SHARED_COST",
  "VOIP",
  "PERSONAL_NUMBER",
  "PAGER",
  "UAN",
  "VOICEMAIL",
  "UNKNOWN"
], Yt = new Set(Rs), qe = {
  COUNTRY_CODE: "countryCode",
  DIAL_CODE: "dialCode"
}, N = {
  EXPANDED: "aria-expanded",
  LABEL: "aria-label",
  SELECTED: "aria-selected",
  ACTIVE_DESCENDANT: "aria-activedescendant",
  HASPOPUP: "aria-haspopup",
  CONTROLS: "aria-controls",
  HIDDEN: "aria-hidden",
  AUTOCOMPLETE: "aria-autocomplete",
  MODAL: "aria-modal"
}, ks = {
  selectedCountryAriaLabel: "Change country, selected ${countryName} (${dialCode})",
  noCountrySelected: "Select country",
  countryListAriaLabel: "List of countries",
  searchPlaceholder: "Search",
  clearSearchAriaLabel: "Clear search",
  searchEmptyState: "No results found",
  searchSummaryAria(t) {
    return t === 0 ? "No results found" : t === 1 ? "1 result found" : `${t} results found`;
  }
}, ue = (t) => typeof window < "u" && typeof window.matchMedia == "function" && window.matchMedia(t).matches, Ms = () => {
  if (typeof navigator < "u" && typeof window < "u") {
    const t = ue("(max-width: 500px)"), e = ue("(max-height: 600px)"), n = ue("(pointer: coarse)");
    return t || n && e;
  }
  return !1;
}, Pe = {
  //* Whether or not to allow the dropdown.
  allowDropdown: !0,
  //* The number type to enforce during validation.
  allowedNumberTypes: ["MOBILE", "FIXED_LINE"],
  //* Whether or not to allow extensions after the main number.
  allowNumberExtensions: !1,
  // Allow alphanumeric "phonewords" (e.g. +1 800 FLOWERS) as valid numbers
  allowPhonewords: !1,
  //* Add a placeholder in the input with an example number for the selected country.
  autoPlaceholder: te.POLITE,
  //* Modify the parentClass.
  containerClass: "",
  //* Locale for localising country names via Intl.DisplayNames.
  countryNameLocale: "en",
  //* The order of the countries in the dropdown. Defaults to alphabetical.
  countryOrder: null,
  //* Add a country search input at the top of the dropdown.
  countrySearch: !0,
  //* Modify the auto placeholder.
  customPlaceholder: null,
  //* Always show the dropdown
  dropdownAlwaysOpen: !1,
  //* Append menu to specified element.
  dropdownContainer: null,
  //* Don't display these countries.
  excludeCountries: [],
  //* Fix the dropdown width to the input width (rather than being as wide as the longest country name).
  fixDropdownWidth: !0,
  //* Format the number as the user types
  formatAsYouType: !0,
  //* Format the input value during initialisation and on setNumber.
  formatOnDisplay: !0,
  //* geoIp lookup function.
  geoIpLookup: null,
  //* Inject a hidden input with the name returned from this function, and on submit, populate it with the result of getNumber.
  hiddenInput: null,
  //* Internationalise the plugin text e.g. search input placeholder, country names.
  i18n: {},
  //* Initial country.
  initialCountry: "",
  //* A function to load the utils script.
  loadUtils: null,
  //* National vs international formatting for numbers e.g. placeholders and displaying existing numbers.
  nationalMode: !0,
  //* Display only these countries.
  onlyCountries: [],
  //* Number type to use for placeholders.
  placeholderNumberType: "MOBILE",
  //* Show flags - for both the selected country, and in the country dropdown
  showFlags: !0,
  //* Display the international dial code next to the selected flag.
  separateDialCode: !1,
  //* Only allow certain chars e.g. a plus followed by numeric digits, and cap at max valid length.
  strictMode: !1,
  //* Use full screen popup instead of dropdown for country list.
  useFullscreenPopup: Ms()
}, Vn = (t) => JSON.stringify(t), Xe = (t) => !!t && typeof t == "object" && !Array.isArray(t), Us = (t) => typeof t == "function", xs = (t) => {
  if (!t || typeof t != "object") return !1;
  const e = t;
  return e.nodeType === 1 && typeof e.tagName == "string" && typeof e.appendChild == "function";
}, Fs = new Set(pt.map((t) => t.iso2)), Wn = (t) => Fs.has(t), Qe = new Set(Object.values(te)), ee = (t) => {
  console.warn(`[intl-tel-input] ${t}`);
}, V = (t, e, n) => {
  ee(`Option '${t}' must be ${e}; got ${Vn(n)}. Ignoring.`);
}, Hs = (t, e) => Object.prototype.hasOwnProperty.call(t, e), Ze = (t, e) => {
  const n = "an array of ISO2 country code strings";
  if (!Array.isArray(e))
    return V(t, n, e), !1;
  for (const i of e) {
    if (typeof i != "string")
      return V(t, n, e), !1;
    const s = i.toLowerCase();
    if (!Wn(s))
      return ee(`Invalid country code in '${t}': '${i}'. Ignoring.`), !1;
  }
  return !0;
}, Bs = (t) => {
  if (t === void 0)
    return {};
  if (!Xe(t)) {
    const n = `The second argument must be an options object; got ${Vn(t)}. Using defaults.`;
    return ee(n), {};
  }
  const e = {};
  for (const [n, i] of Object.entries(t)) {
    if (!Hs(Pe, n)) {
      ee(`Unknown option '${n}'. Ignoring.`);
      continue;
    }
    switch (n) {
      case "allowDropdown":
      case "allowNumberExtensions":
      case "allowPhonewords":
      case "countrySearch":
      case "dropdownAlwaysOpen":
      case "fixDropdownWidth":
      case "formatAsYouType":
      case "formatOnDisplay":
      case "nationalMode":
      case "showFlags":
      case "separateDialCode":
      case "strictMode":
      case "useFullscreenPopup":
        if (typeof i != "boolean") {
          V(n, "a boolean", i);
          break;
        }
        e[n] = i;
        break;
      case "autoPlaceholder":
        if (typeof i != "string" || !Qe.has(i)) {
          const s = Array.from(Qe).join(", ");
          V("autoPlaceholder", `one of ${s}`, i);
          break;
        }
        e[n] = i;
        break;
      case "containerClass":
      case "countryNameLocale":
        if (typeof i != "string") {
          V(n, "a string", i);
          break;
        }
        e[n] = i;
        break;
      case "countryOrder":
        (i === null || Ze(n, i)) && (e[n] = i);
        break;
      case "customPlaceholder":
      case "geoIpLookup":
      case "hiddenInput":
      case "loadUtils":
        if (i !== null && !Us(i)) {
          V(n, "a function or null", i);
          break;
        }
        e[n] = i;
        break;
      case "dropdownContainer":
        if (i !== null && !xs(i)) {
          V("dropdownContainer", "an HTMLElement or null", i);
          break;
        }
        e[n] = i;
        break;
      case "excludeCountries":
      case "onlyCountries":
        Ze(n, i) && (e[n] = i);
        break;
      case "i18n":
        if (i && !Xe(i)) {
          V("i18n", "an object", i);
          break;
        }
        e[n] = i;
        break;
      case "initialCountry": {
        if (typeof i != "string") {
          V("initialCountry", "a string", i);
          break;
        }
        const s = i.toLowerCase();
        if (s && s !== Mt.AUTO && !Wn(s)) {
          V("initialCountry", "a valid ISO2 country code or 'auto'", i);
          break;
        }
        e[n] = i;
        break;
      }
      case "placeholderNumberType":
        if (typeof i != "string" || !Yt.has(i)) {
          const s = Array.from(Yt).join(", ");
          V("placeholderNumberType", `one of ${s}`, i);
          break;
        }
        e[n] = i;
        break;
      case "allowedNumberTypes":
        if (i !== null) {
          if (!Array.isArray(i)) {
            V("allowedNumberTypes", "an array of number types or null", i);
            break;
          }
          let s = !0;
          for (const r of i)
            if (typeof r != "string" || !Yt.has(r)) {
              const o = Array.from(Yt).join(", ");
              V("allowedNumberTypes", `an array of valid number types (${o})`, r), s = !1;
              break;
            }
          s && (e[n] = i);
        } else
          e[n] = null;
        break;
    }
  }
  return e;
}, Vs = (t) => {
  t.dropdownAlwaysOpen && (t.useFullscreenPopup = !1, t.allowDropdown = !0), t.useFullscreenPopup && (t.fixDropdownWidth = !1), t.onlyCountries.length === 1 && (t.initialCountry = t.onlyCountries[0]), t.separateDialCode && (t.nationalMode = !1), t.allowDropdown && !t.showFlags && !t.separateDialCode && (t.nationalMode = !1), t.useFullscreenPopup && !t.dropdownContainer && (t.dropdownContainer = document.body), t.i18n = { ...ks, ...t.i18n };
}, zt = (t) => t.replace(/\D/g, ""), jn = (t = "") => t.normalize("NFD").replace(/[\u0300-\u036f]/g, "").toLowerCase(), Ws = () => typeof navigator < "u" ? /Android/i.test(navigator.userAgent) : !1, js = (t, e) => {
  const n = jn(e), i = [], s = [], r = [], o = [], l = [], a = [];
  for (const c of t)
    c.iso2 === n ? i.push(c) : c.normalisedName.startsWith(n) ? s.push(c) : c.normalisedName.includes(n) ? r.push(c) : n === c.dialCode || n === c.dialCodePlus ? o.push(c) : c.dialCodePlus.includes(n) ? l.push(c) : c.initials.includes(n) && a.push(c);
  const u = (c, d) => c.priority - d.priority;
  return [
    ...i.sort(u),
    ...s.sort(u),
    ...r.sort(u),
    ...o.sort(u),
    ...l.sort(u),
    ...a.sort(u)
  ];
}, Gs = (t, e) => {
  const n = e.toLowerCase();
  for (const i of t)
    if (i.name.toLowerCase().startsWith(n))
      return i;
  return null;
}, ce = (t) => Object.keys(t).filter((e) => !!t[e]).join(" "), L = (t, e, n) => {
  const i = document.createElement(t);
  return e && Object.entries(e).forEach(
    ([s, r]) => i.setAttribute(s, r)
  ), n && n.appendChild(i), i;
}, Ys = () => `
  <svg class="iti__search-icon-svg" width="14" height="14" viewBox="0 0 24 24" focusable="false" ${N.HIDDEN}="true">
    <circle cx="11" cy="11" r="7" />
    <line x1="21" y1="21" x2="16.65" y2="16.65" />
  </svg>`, $s = (t) => {
  const e = `iti-${t}-clear-mask`;
  return `
    <svg class="iti__search-clear-svg" width="12" height="12" viewBox="0 0 16 16" ${N.HIDDEN}="true" focusable="false">
      <mask id="${e}" maskUnits="userSpaceOnUse">
        <rect width="16" height="16" fill="white" />
        <path d="M5.2 5.2 L10.8 10.8 M10.8 5.2 L5.2 10.8" stroke="black" stroke-linecap="round" class="iti__search-clear-x" />
      </mask>
      <circle cx="8" cy="8" r="8" class="iti__search-clear-bg" mask="url(#${e})" />
    </svg>`;
}, Ks = () => `
  <svg class="iti__country-check-svg" width="14" height="14" viewBox="0 0 16 16" fill="currentColor" focusable="false" ${N.HIDDEN}="true">
    <path d="M16 8A8 8 0 1 1 0 8a8 8 0 0 1 16 0m-3.97-3.03a.75.75 0 0 0-1.08.022L7.477 9.417 5.384 7.323a.75.75 0 0 0-1.06 1.06L6.97 11.03a.75.75 0 0 0 1.079-.02l3.992-4.99a.75.75 0 0 0-.01-1.05z"/>
  </svg>`, Je = () => `
  <svg width="256" height="256" viewBox="0 0 512 512" class="iti__globe-svg">
    <path d="M508 213a240 240 0 0 0-449-87l-2 5-2 5c-8 14-13 30-17 46a65 65 0 0 1 56 4c16-10 35-19 56-27l9-3c-6 23-10 48-10 74h-16l4 6c3 4 5 8 6 13h6c0 22 3 44 8 65l2 10-25-10-4 5 12 18 9 3 6 2 8 3 9 26 1 2 16-7h1l-5-13-1-2c24 6 49 9 75 10v26l11 10 7 7v-30l1-13c22 0 44-3 65-8l10-2-21 48-1 1a317 317 0 0 1-14 23l-21 5h-2c6 16 7 33 1 50a240 240 0 0 0 211-265m-401-56-11 6c19-44 54-79 98-98-11 20-21 44-29 69-21 6-40 15-58 23m154 182v4c-29-1-57-6-81-13-7-25-12-52-13-81h94zm0-109h-94c1-29 6-56 13-81 24-7 52-12 81-13zm0-112c-22 1-44 4-65 8l-10 2 12-30 9-17 1-2a332 332 0 0 1 13-23c13-4 26-6 40-7zm187 69 6 4c4 12 6 25 6 38v1h-68c-1-26-4-51-10-74l48 20 1 1 14 8zm-14-44 10 20c-20-11-43-21-68-29-8-25-18-49-29-69 37 16 67 44 87 78M279 49h1c13 1 27 3 39 7l14 23 1 2a343 343 0 0 1 12 26l2 5 6 16c-23-6-48-9-74-10h-1zm0 87h1c29 1 56 6 81 13 7 24 12 51 12 80v1h-94zm2 207h-2v-94h95c-1 29-6 56-13 81-24 7-51 12-80 13m86 60-20 10c11-20 21-43 29-68 25-8 48-18 68-29-16 37-43 67-77 87m87-115-7 5-16 9-2 1a337 337 0 0 1-47 21c6-24 9-49 10-75h68c0 13-2 27-6 39"/>
    <path d="m261 428-2-2-22-21a40 40 0 0 0-32-11h-1a37 37 0 0 0-18 8l-1 1-4 2-2 2-5 4c-9-3-36-31-47-44s-32-45-34-55l3-2a151 151 0 0 0 11-9v-1a39 39 0 0 0 5-48l-3-3-11-19-3-4-5-7h-1l-3-3-4-3-5-2a35 35 0 0 0-16-3h-5c-4 1-14 5-24 11l-4 2-4 3-4 2c-9 8-17 17-18 27a380 380 0 0 0 212 259h3c12 0 25-10 36-21l10-12 6-11a39 39 0 0 0-8-40"/>
  </svg>`;
class zs {
  constructor(e, n, i) {
    this.highlightedItem = null, this.selectedItem = null, e.dataset.intlTelInputId = i.toString(), this.telInput = e, this.options = n, this.id = i, this.hadInitialPlaceholder = !!e.getAttribute("placeholder"), this.isRTL = !!this.telInput.closest("[dir=rtl]"), this.options.separateDialCode && (this.originalPaddingLeft = this.telInput.style.paddingLeft);
  }
  //* Generate all of the markup for the plugin: the selected country overlay, and the dropdown.
  generateMarkup(e) {
    this.countries = e, this.telInput.classList.add("iti__tel-input"), this.telInput.hasAttribute("autocomplete") || this.telInput.setAttribute("autocomplete", "tel"), this.telInput.hasAttribute("inputmode") || this.telInput.setAttribute("inputmode", "tel");
    const n = this._createWrapperAndInsert();
    this._maybeBuildCountryContainer(n), n.appendChild(this.telInput), this._maybeUpdateInputPaddingAndReveal(), this._maybeBuildHiddenInputs(n);
  }
  _createWrapperAndInsert() {
    const { allowDropdown: e, showFlags: n, containerClass: i, useFullscreenPopup: s } = this.options, r = ce({
      iti: !0,
      "iti--allow-dropdown": e,
      "iti--show-flags": n,
      "iti--inline-dropdown": !s,
      [i]: !!i
    }), o = L("div", { class: r });
    return this.isRTL && o.setAttribute("dir", "ltr"), this.telInput.before(o), o;
  }
  _maybeBuildCountryContainer(e) {
    const { allowDropdown: n, separateDialCode: i, showFlags: s } = this.options;
    if (n || s || i) {
      this.countryContainer = L(
        "div",
        // visibly hidden until we measure it's width to set the input padding correctly
        { class: `iti__country-container ${I.V_HIDE}` },
        e
      ), n ? (this.selectedCountry = L(
        "button",
        {
          type: "button",
          class: "iti__selected-country",
          [N.EXPANDED]: "false",
          [N.LABEL]: this.options.i18n.noCountrySelected,
          [N.HASPOPUP]: "dialog",
          [N.CONTROLS]: `iti-${this.id}__dropdown-content`
        },
        this.countryContainer
      ), this.telInput.disabled && this.selectedCountry.setAttribute("disabled", "true")) : this.selectedCountry = L(
        "div",
        { class: "iti__selected-country" },
        this.countryContainer
      );
      const r = L(
        "div",
        { class: "iti__selected-country-primary" },
        this.selectedCountry
      );
      this.selectedCountryInner = L(
        "div",
        { class: I.FLAG },
        r
      ), n && (this.dropdownArrow = L(
        "div",
        { class: "iti__arrow", [N.HIDDEN]: "true" },
        r
      )), i && (this.selectedDialCode = L(
        "div",
        { class: "iti__selected-dial-code" },
        this.selectedCountry
      )), n && this._buildDropdownContent();
    }
  }
  _buildDropdownContent() {
    const {
      fixDropdownWidth: e,
      useFullscreenPopup: n,
      countrySearch: i,
      i18n: s,
      dropdownContainer: r,
      containerClass: o
    } = this.options, l = e ? "" : "iti--flexible-dropdown-width";
    if (this.dropdownContent = L("div", {
      id: `iti-${this.id}__dropdown-content`,
      class: `iti__dropdown-content ${I.HIDE} ${l}`,
      role: "dialog",
      [N.MODAL]: "true"
    }), this.isRTL && this.dropdownContent.setAttribute("dir", "rtl"), i && this._buildSearchUI(), this.countryList = L(
      "ul",
      {
        class: "iti__country-list",
        id: `iti-${this.id}__country-listbox`,
        role: "listbox",
        [N.LABEL]: s.countryListAriaLabel
      },
      this.dropdownContent
    ), this._appendListItems(), i && this.updateSearchResultsA11yText(), r) {
      const a = ce({
        iti: !0,
        "iti--container": !0,
        "iti--fullscreen-popup": n,
        "iti--inline-dropdown": !n,
        [o]: !!o
      });
      this.dropdown = L("div", { class: a }), this.dropdown.appendChild(this.dropdownContent);
    } else
      this.countryContainer.appendChild(this.dropdownContent);
  }
  _buildSearchUI() {
    const { i18n: e } = this.options, n = L(
      "div",
      { class: "iti__search-input-wrapper" },
      this.dropdownContent
    );
    this.searchIcon = L(
      "span",
      {
        class: "iti__search-icon",
        [N.HIDDEN]: "true"
      },
      n
    ), this.searchIcon.innerHTML = Ys(), this.searchInput = L(
      "input",
      {
        id: `iti-${this.id}__search-input`,
        // Chrome says inputs need either a name or an id
        type: "search",
        class: "iti__search-input",
        placeholder: e.searchPlaceholder,
        // role=combobox + aria-autocomplete=list + aria-activedescendant allows maintaining focus on the search input while allowing users to navigate search results with up/down keyboard keys
        role: "combobox",
        [N.EXPANDED]: "true",
        [N.LABEL]: e.searchPlaceholder,
        [N.CONTROLS]: `iti-${this.id}__country-listbox`,
        [N.AUTOCOMPLETE]: "list",
        autocomplete: "off"
      },
      n
    ), this.searchClearButton = L(
      "button",
      {
        type: "button",
        class: `iti__search-clear ${I.HIDE}`,
        [N.LABEL]: e.clearSearchAriaLabel,
        tabindex: "-1"
      },
      n
    ), this.searchClearButton.innerHTML = $s(this.id), this.searchResultsA11yText = L(
      "span",
      { class: "iti__a11y-text" },
      this.dropdownContent
    ), this.searchNoResults = L(
      "div",
      {
        class: `iti__no-results ${I.HIDE}`,
        [N.HIDDEN]: "true"
        // all a11y messaging happens in this.searchResultsA11yText
      },
      this.dropdownContent
    ), this.searchNoResults.textContent = e.searchEmptyState;
  }
  _maybeUpdateInputPaddingAndReveal() {
    this.countryContainer && (this.updateInputPadding(), this.countryContainer.classList.remove(I.V_HIDE));
  }
  _maybeBuildHiddenInputs(e) {
    var i, s;
    const { hiddenInput: n } = this.options;
    if (n) {
      const r = this.telInput.getAttribute("name") || "", o = n(r);
      if (o.phone) {
        const l = (i = this.telInput.form) == null ? void 0 : i.querySelector(
          `input[name="${o.phone}"]`
        );
        l ? this.hiddenInput = l : (this.hiddenInput = L("input", {
          type: "hidden",
          name: o.phone
        }), e.appendChild(this.hiddenInput));
      }
      if (o.country) {
        const l = (s = this.telInput.form) == null ? void 0 : s.querySelector(
          `input[name="${o.country}"]`
        );
        l ? this.hiddenInputCountry = l : (this.hiddenInputCountry = L("input", {
          type: "hidden",
          name: o.country
        }), e.appendChild(this.hiddenInputCountry));
      }
    }
  }
  //* For each country: add a country list item <li> to the countryList <ul> container.
  _appendListItems() {
    const e = document.createDocumentFragment();
    for (let n = 0; n < this.countries.length; n++) {
      const i = this.countries[n], s = ce({
        [I.COUNTRY_ITEM]: !0
      }), r = L("li", {
        id: `iti-${this.id}__item-${i.iso2}`,
        class: s,
        tabindex: "-1",
        role: "option",
        [N.SELECTED]: "false"
      });
      r.dataset.dialCode = i.dialCode, r.dataset.countryCode = i.iso2, i.nodeById[this.id] = r, this.options.showFlags && L("div", { class: `${I.FLAG} iti__${i.iso2}` }, r);
      const o = L("span", { class: "iti__country-name" }, r);
      o.textContent = i.name;
      const l = L("span", { class: "iti__dial-code" }, o);
      this.isRTL && l.setAttribute("dir", "ltr"), l.textContent = `+${i.dialCode}`, e.appendChild(r);
    }
    this.countryList.appendChild(e);
  }
  //* Update the input padding to make space for the selected country/dial code.
  updateInputPadding() {
    if (this.selectedCountry) {
      const e = this.options.separateDialCode ? le.SANE_SELECTED_WITH_DIAL_WIDTH : le.SANE_SELECTED_NO_DIAL_WIDTH, i = (this.selectedCountry.offsetWidth || this._getHiddenSelectedCountryWidth() || e) + le.INPUT_PADDING_EXTRA_LEFT;
      this.telInput.style.paddingLeft = `${i}px`;
    }
  }
  //* When input is in a hidden container during init, we cannot calculate the selected country width.
  //* Fix: clone the markup, make it invisible, add it to the end of the DOM, and then measure it's width.
  //* To get the right styling to apply, all we need is a shallow clone of the container,
  //* and then to inject a deep clone of the selectedCountry element.
  _getHiddenSelectedCountryWidth() {
    if (this.telInput.parentNode) {
      let e;
      try {
        e = window.top.document.body;
      } catch {
        e = document.body;
      }
      const n = this.telInput.parentNode.cloneNode(
        !1
      );
      n.style.visibility = "hidden", e.appendChild(n);
      const i = this.countryContainer.cloneNode();
      n.appendChild(i);
      const s = this.selectedCountry.cloneNode(
        !0
      );
      i.appendChild(s);
      const r = s.offsetWidth;
      return e.removeChild(n), r;
    }
    return 0;
  }
  //* Update search results text (for a11y).
  updateSearchResultsA11yText() {
    const { i18n: e } = this.options, n = this.countryList.childElementCount;
    this.searchResultsA11yText.textContent = e.searchSummaryAria(n);
  }
  //* Check if an element is visible within it's container, else scroll until it is.
  scrollTo(e) {
    const n = this.countryList, i = document.documentElement.scrollTop, s = n.offsetHeight, r = n.getBoundingClientRect().top + i, o = r + s, l = e.offsetHeight, a = e.getBoundingClientRect().top + i, u = a + l, c = a - r + n.scrollTop;
    if (a < r)
      n.scrollTop = c;
    else if (u > o) {
      const d = s - l;
      n.scrollTop = c - d;
    }
  }
  //* Remove highlighting from the previous list item and highlight the new one.
  highlightListItem(e, n) {
    const i = this.highlightedItem;
    if (i && i.classList.remove(I.HIGHLIGHT), this.highlightedItem = e, this.highlightedItem && (this.highlightedItem.classList.add(I.HIGHLIGHT), this.options.countrySearch)) {
      const s = this.highlightedItem.getAttribute("id") || "";
      this.searchInput.setAttribute(N.ACTIVE_DESCENDANT, s);
    }
    n && this.highlightedItem.focus();
  }
  // Update the selected list item in the dropdown
  updateSelectedItem(e) {
    var n;
    if (this.selectedItem && this.selectedItem.dataset.countryCode !== e && (this.selectedItem.setAttribute(N.SELECTED, "false"), (n = this.selectedItem.querySelector(".iti__country-check")) == null || n.remove(), this.selectedItem = null), e && !this.selectedItem) {
      const i = this.countryList.querySelector(
        `[data-country-code="${e}"]`
      );
      if (i) {
        i.setAttribute(N.SELECTED, "true");
        const s = L(
          "span",
          { class: "iti__country-check", [N.HIDDEN]: "true" },
          i
        );
        s.innerHTML = Ks(), this.selectedItem = i;
      }
    }
  }
  //* Country search: Filter the country list to the given array of countries.
  filterCountries(e) {
    this.countryList.innerHTML = "";
    let n = !0;
    for (const i of e) {
      const s = i.nodeById[this.id];
      s && (this.countryList.appendChild(s), n && (this.highlightListItem(s, !1), n = !1));
    }
    n ? (this.highlightListItem(null, !1), this.searchNoResults && this.searchNoResults.classList.remove(I.HIDE)) : this.searchNoResults && this.searchNoResults.classList.add(I.HIDE), this.countryList.scrollTop = 0, this.updateSearchResultsA11yText();
  }
  destroy() {
    this.telInput.iti = void 0, delete this.telInput.dataset.intlTelInputId, this.options.separateDialCode && (this.telInput.style.paddingLeft = this.originalPaddingLeft);
    const e = this.telInput.parentNode;
    e.before(this.telInput), e.remove(), this.telInput = null, this.countryContainer = null, this.selectedCountry = null, this.selectedCountryInner = null, this.selectedDialCode = null, this.dropdownArrow = null, this.dropdownContent = null, this.searchInput = null, this.searchIcon = null, this.searchClearButton = null, this.searchNoResults = null, this.searchResultsA11yText = null, this.countryList = null, this.dropdown = null, this.hiddenInput = null, this.hiddenInputCountry = null, this.highlightedItem = null, this.selectedItem = null;
    for (const n of this.countries)
      delete n.nodeById[this.id];
    this.countries = null;
  }
}
const qs = (t) => {
  const { onlyCountries: e, excludeCountries: n } = t;
  if (e != null && e.length) {
    const i = e.map(
      (s) => s.toLowerCase()
    );
    return pt.filter(
      (s) => i.includes(s.iso2)
    );
  } else if (n != null && n.length) {
    const i = n.map(
      (s) => s.toLowerCase()
    );
    return pt.filter(
      (s) => !i.includes(s.iso2)
    );
  }
  return pt;
}, Xs = (t, e) => {
  const { countryNameLocale: n, i18n: i } = e;
  let s;
  try {
    typeof Intl < "u" && typeof Intl.DisplayNames == "function" ? s = new Intl.DisplayNames(n, {
      type: "region"
    }) : s = null;
  } catch (r) {
    console.error(r), s = null;
  }
  for (const r of t)
    r.name = i[r.iso2] || (s == null ? void 0 : s.of(r.iso2.toUpperCase())) || "";
}, Qs = (t) => {
  const e = /* @__PURE__ */ new Set();
  let n = 0;
  const i = {}, s = (o, l) => {
    if (!o || !l)
      return;
    l.length > n && (n = l.length), i.hasOwnProperty(l) || (i[l] = []);
    const a = i[l];
    a.includes(o) || a.push(o);
  }, r = [...t].sort((o, l) => o.priority - l.priority);
  for (const o of r) {
    e.has(o.dialCode) || e.add(o.dialCode);
    for (let l = 1; l < o.dialCode.length; l++) {
      const a = o.dialCode.substring(0, l);
      s(o.iso2, a);
    }
    if (s(o.iso2, o.dialCode), o.areaCodes) {
      const l = i[o.dialCode][0];
      for (const a of o.areaCodes) {
        for (let u = 1; u < a.length; u++) {
          const c = a.substring(0, u), d = o.dialCode + c;
          s(l, d), s(o.iso2, d);
        }
        s(o.iso2, o.dialCode + a);
      }
    }
  }
  return { dialCodes: e, dialCodeMaxLen: n, dialCodeToIso2Map: i };
}, Zs = (t, e) => {
  e.countryOrder && (e.countryOrder = e.countryOrder.map(
    (n) => n.toLowerCase()
  )), t.sort((n, i) => {
    const { countryOrder: s } = e;
    if (s) {
      const r = s.indexOf(n.iso2), o = s.indexOf(i.iso2), l = r > -1, a = o > -1;
      if (l || a)
        return l && a ? r - o : l ? -1 : 1;
    }
    return n.name.localeCompare(i.name);
  });
}, Js = (t) => {
  for (const e of t)
    e.normalisedName = jn(e.name), e.initials = e.normalisedName.split(/[^a-z]/).map((n) => n[0]).join(""), e.dialCodePlus = `+${e.dialCode}`;
}, tn = (t, e, n, i) => {
  let s = t;
  if (n && e) {
    const r = `+${i.dialCode}`, o = s[r.length] === " " || s[r.length] === "-" ? r.length + 1 : r.length;
    s = s.substring(o);
  }
  return s;
}, tr = (t, e, n, i, s) => {
  const r = n ? n.formatNumberAsYouType(t, i.iso2) : t, { dialCode: o } = i;
  return s && e.charAt(0) !== "+" && r.includes(`+${o}`) ? (r.split(`+${o}`)[1] || "").trim() : r;
}, er = (t, e, n, i) => {
  if (n === 0 && !i)
    return 0;
  let s = 0;
  for (let r = 0; r < e.length; r++) {
    if (/[+0-9]/.test(e[r]) && s++, s === t && !i)
      return r + 1;
    if (i && s === t + 1)
      return r;
  }
  return e.length;
}, nr = [
  "800",
  "822",
  "833",
  "844",
  "855",
  "866",
  "877",
  "880",
  "881",
  "882",
  "883",
  "884",
  "885",
  "886",
  "887",
  "888",
  "889"
], en = (t) => {
  const e = zt(t);
  if (e.startsWith(Bn.NANP) && e.length >= 4) {
    const n = e.substring(1, 4);
    return nr.includes(n);
  }
  return !1;
};
let ir = 0;
const sr = new Set(pt.map((t) => t.iso2)), de = (t) => sr.has(t), rr = (t) => {
  const e = t == null ? void 0 : t.tagName;
  if (!(!!t && typeof t == "object" && e === "INPUT" && typeof t.setAttribute == "function")) {
    const i = Object.prototype.toString.call(t);
    throw new TypeError(
      `The first argument must be an HTMLInputElement, not ${i}`
    );
  }
};
class or {
  constructor(e, n = {}) {
    this.id = ir++, rr(e);
    const i = Bs(n);
    this.options = { ...Pe, ...i }, Vs(this.options), this.ui = new zs(e, this.options, this.id), this.isAndroid = Ws(), this.promise = this._createInitPromises(this.options), this.countries = qs(this.options);
    const { dialCodes: s, dialCodeMaxLen: r, dialCodeToIso2Map: o } = Qs(
      this.countries
    );
    this.dialCodes = s, this.dialCodeMaxLen = r, this.dialCodeToIso2Map = o, this.countryByIso2 = new Map(this.countries.map((l) => [l.iso2, l])), this._init();
  }
  _updateNumeralSet(e) {
    /[\u0660-\u0669]/.test(e) ? this.userNumeralSet = "arabic-indic" : /[\u06F0-\u06F9]/.test(e) ? this.userNumeralSet = "persian" : this.userNumeralSet = "ascii";
  }
  _mapAsciiToUserNumerals(e) {
    if (this.userNumeralSet || this._updateNumeralSet(this.ui.telInput.value), this.userNumeralSet === "ascii")
      return e;
    const n = this.userNumeralSet === "arabic-indic" ? 1632 : 1776;
    return e.replace(/[0-9]/g, (i) => String.fromCharCode(n + Number(i)));
  }
  // Normalize Eastern Arabic (U+0660-0669) and Persian/Extended Arabic-Indic (U+06F0-06F9) numerals to ASCII 0-9
  _normaliseNumerals(e) {
    if (!e)
      return "";
    if (this._updateNumeralSet(e), this.userNumeralSet === "ascii")
      return e;
    const n = this.userNumeralSet === "arabic-indic" ? 1632 : 1776, i = this.userNumeralSet === "arabic-indic" ? /[\u0660-\u0669]/g : /[\u06F0-\u06F9]/g;
    return e.replace(i, (s) => String.fromCharCode(48 + (s.charCodeAt(0) - n)));
  }
  _getTelInputValue() {
    const e = this.ui.telInput.value.trim();
    return this._normaliseNumerals(e);
  }
  _setTelInputValue(e) {
    this.ui.telInput.value = this._mapAsciiToUserNumerals(e);
  }
  _createInitPromises(e) {
    const { initialCountry: n, geoIpLookup: i, loadUtils: s } = e, r = n === Mt.AUTO && !!i, o = !!s && !C.utils;
    let l;
    r ? l = new Promise((u, c) => {
      this.resolveAutoCountryPromise = u, this.rejectAutoCountryPromise = c;
    }) : (l = Promise.resolve(void 0), this.resolveAutoCountryPromise = () => {
    }, this.rejectAutoCountryPromise = () => {
    });
    let a;
    return o ? a = new Promise((u, c) => {
      this.resolveUtilsScriptPromise = u, this.rejectUtilsScriptPromise = c;
    }) : (a = Promise.resolve(void 0), this.resolveUtilsScriptPromise = () => {
    }, this.rejectUtilsScriptPromise = () => {
    }), Promise.all([l, a]);
  }
  //* Can't be private as it's called from intlTelInput convenience wrapper.
  _init() {
    this.selectedCountryData = {}, this.abortController = new AbortController(), this._processCountryData(), this.ui.generateMarkup(this.countries), this._setInitialState(), this._initListeners(), this._initRequests(), this.options.dropdownAlwaysOpen && this._openDropdown();
  }
  //********************
  //*  PRIVATE METHODS
  //********************
  //* Prepare all of the country data, including onlyCountries, excludeCountries, countryOrder options.
  _processCountryData() {
    Xs(this.countries, this.options), Zs(this.countries, this.options), Js(this.countries);
  }
  //* Set the initial state of the input value and the selected country by:
  //* 1. Extracting a dial code from the given number
  //* 2. Using explicit initialCountry
  _setInitialState(e = !1) {
    const n = this.ui.telInput.getAttribute("value"), i = this._normaliseNumerals(n), s = this._getTelInputValue(), o = i && i.startsWith("+") && (!s || !s.startsWith("+")) ? i : s, l = this._getDialCode(o), a = en(o), { initialCountry: u, geoIpLookup: c } = this.options, f = u === Mt.AUTO && c && !e, h = u.toLowerCase(), _ = de(h);
    l ? a ? _ ? this._setCountry(h) : f || this._setCountry(Os.ISO2) : this._updateCountryFromNumber(o) : _ ? this._setCountry(h) : f || this._setCountry(""), o && this._updateValFromNumber(o);
  }
  //* Initialise the main event listeners: input keyup, and click selected country.
  _initListeners() {
    this._initTelInputListeners(), this.options.allowDropdown && this._initDropdownListeners(), (this.ui.hiddenInput || this.ui.hiddenInputCountry) && this.ui.telInput.form && this._initHiddenInputListener();
  }
  //* Update hidden input on form submit.
  _initHiddenInputListener() {
    var n;
    const e = () => {
      this.ui.hiddenInput && (this.ui.hiddenInput.value = this.getNumber()), this.ui.hiddenInputCountry && (this.ui.hiddenInputCountry.value = this.selectedCountryData.iso2 || "");
    };
    (n = this.ui.telInput.form) == null || n.addEventListener("submit", e, {
      signal: this.abortController.signal
    });
  }
  //* initialise the dropdown listeners.
  _initDropdownListeners() {
    const e = this.abortController.signal, n = (o) => {
      this.ui.dropdownContent.classList.contains(I.HIDE) ? this.ui.telInput.focus() : o.preventDefault();
    }, i = this.ui.telInput.closest("label");
    i && i.addEventListener("click", n, { signal: e });
    const s = () => {
      this.ui.dropdownContent.classList.contains(
        I.HIDE
      ) && !this.ui.telInput.disabled && !this.ui.telInput.readOnly && this._openDropdown();
    };
    this.ui.selectedCountry.addEventListener(
      "click",
      s,
      {
        signal: e
      }
    );
    const r = (o) => {
      this.ui.dropdownContent.classList.contains(
        I.HIDE
      ) && [M.ARROW_UP, M.ARROW_DOWN, M.SPACE, M.ENTER].includes(o.key) && (o.preventDefault(), o.stopPropagation(), this._openDropdown()), o.key === M.TAB && this._closeDropdown();
    };
    this.ui.countryContainer.addEventListener(
      "keydown",
      r,
      { signal: e }
    );
  }
  //* Init requests: utils script / geo ip lookup.
  _initRequests() {
    const { loadUtils: e, initialCountry: n, geoIpLookup: i } = this.options;
    if (e && !C.utils) {
      const r = () => {
        var o;
        (o = C.attachUtils(e)) == null || o.catch(() => {
        });
      };
      if (C.documentReady())
        r();
      else {
        const o = () => {
          r();
        };
        window.addEventListener("load", o, {
          signal: this.abortController.signal
        });
      }
    } else
      this.resolveUtilsScriptPromise();
    n === Mt.AUTO && i && (this.selectedCountryData.iso2 ? this.resolveAutoCountryPromise() : this._loadAutoCountry());
  }
  //* Perform the geo ip lookup.
  _loadAutoCountry() {
    if (C.autoCountry)
      this.handleAutoCountry();
    else if (this.ui.selectedCountryInner.classList.add(I.LOADING), !C.startedLoadingAutoCountry && (C.startedLoadingAutoCountry = !0, typeof this.options.geoIpLookup == "function")) {
      const e = (i = "") => {
        this.ui.selectedCountryInner.classList.remove(I.LOADING);
        const s = i.toLowerCase();
        de(s) ? (C.autoCountry = s, setTimeout(() => Ut("handleAutoCountry"))) : Ut("handleAutoCountryFailure");
      }, n = () => {
        this.ui.selectedCountryInner.classList.remove(I.LOADING), Ut("handleAutoCountryFailure");
      };
      this.options.geoIpLookup(e, n);
    }
  }
  _openDropdownWithPlus() {
    this._openDropdown(), this.ui.searchInput.value = "+", this._filterCountriesByQuery("");
  }
  //* Initialize the tel input listeners.
  _initTelInputListeners() {
    this._bindInputListener(), this._maybeBindKeydownListener(), this._maybeBindPasteListener();
  }
  _bindInputListener() {
    const {
      strictMode: e,
      formatAsYouType: n,
      separateDialCode: i,
      allowDropdown: s,
      countrySearch: r
    } = this.options;
    let o = !1;
    ut.ALPHA_UNICODE.test(this._getTelInputValue()) && (o = !0);
    const l = (a) => {
      const u = this._getTelInputValue();
      if (this.isAndroid && (a == null ? void 0 : a.data) === "+" && i && s && r) {
        const _ = this.ui.telInput.selectionStart || 0, g = u.substring(0, _ - 1), b = u.substring(_);
        this._setTelInputValue(g + b), this._openDropdownWithPlus();
        return;
      }
      this._updateCountryFromNumber(u) && this._triggerCountryChange();
      const c = (a == null ? void 0 : a.data) && ut.NON_PLUS_NUMERIC.test(a.data), d = (a == null ? void 0 : a.inputType) === Ke.PASTE && u;
      c || d && !e ? o = !0 : ut.NON_PLUS_NUMERIC.test(u) || (o = !1);
      const f = (a == null ? void 0 : a.detail) && a.detail.isSetNumber, h = this.userNumeralSet === "ascii";
      if (n && !o && !f && h) {
        const _ = this.ui.telInput.selectionStart || 0, b = u.substring(
          0,
          _
        ).replace(
          ut.NON_PLUS_NUMERIC_GLOBAL,
          ""
        ).length, y = (a == null ? void 0 : a.inputType) === Ke.DELETE_FWD, m = this._getFullNumber(), p = tr(
          m,
          u,
          C.utils,
          this.selectedCountryData,
          i
        ), v = er(
          b,
          p,
          _,
          y
        );
        this._setTelInputValue(p), this.ui.telInput.setSelectionRange(v, v);
      }
      if (i && u.startsWith("+") && this.selectedCountryData.dialCode) {
        const _ = tn(u, !0, i, this.selectedCountryData);
        this._setTelInputValue(_);
      }
    };
    this.ui.telInput.addEventListener(
      "input",
      l,
      {
        signal: this.abortController.signal
      }
    );
  }
  _maybeBindKeydownListener() {
    const { strictMode: e, separateDialCode: n, allowDropdown: i, countrySearch: s } = this.options;
    if (e || n) {
      const r = (o) => {
        if (o.key && o.key.length === 1 && !o.altKey && !o.ctrlKey && !o.metaKey) {
          if (n && i && s && o.key === "+") {
            o.preventDefault(), this._openDropdownWithPlus();
            return;
          }
          if (e) {
            const l = this._getTelInputValue(), u = !l.startsWith("+") && this.ui.telInput.selectionStart === 0 && o.key === "+", c = this._normaliseNumerals(o.key), d = /^[0-9]$/.test(c), f = n ? d : u || d, h = this.ui.telInput, _ = h.selectionStart, g = h.selectionEnd, b = l.slice(0, _), y = l.slice(g), m = b + o.key + y, p = this._getFullNumber(m), v = C.utils.getCoreNumber(
              p,
              this.selectedCountryData.iso2
            ), S = this.maxCoreNumberLength && v.length > this.maxCoreNumberLength, O = this._getNewCountryFromNumber(p) !== null;
            (!f || S && !O && !u) && o.preventDefault();
          }
        }
      };
      this.ui.telInput.addEventListener("keydown", r, {
        signal: this.abortController.signal
      });
    }
  }
  _maybeBindPasteListener() {
    if (this.options.strictMode) {
      const e = (n) => {
        n.preventDefault();
        const i = this.ui.telInput, s = i.selectionStart, r = i.selectionEnd, o = this._getTelInputValue(), l = o.slice(0, s), a = o.slice(r), u = this.selectedCountryData.iso2, c = n.clipboardData.getData("text"), d = this._normaliseNumerals(c), f = s === 0 && r > 0, h = !o.startsWith("+") || f, _ = d.replace(ut.NON_PLUS_NUMERIC_GLOBAL, ""), g = _.startsWith("+"), b = _.replace(/\+/g, ""), y = g && h ? `+${b}` : b;
        let m = l + y + a;
        if (m.length > 5) {
          let v = C.utils.getCoreNumber(m, u);
          for (; v.length === 0 && m.length > 0; )
            m = m.slice(0, -1), v = C.utils.getCoreNumber(m, u);
          if (!v)
            return;
          if (this.maxCoreNumberLength && v.length > this.maxCoreNumberLength)
            if (i.selectionEnd === o.length) {
              const S = v.length - this.maxCoreNumberLength;
              m = m.slice(0, m.length - S);
            } else
              return;
        }
        this._setTelInputValue(m);
        const p = s + y.length;
        i.setSelectionRange(p, p), i.dispatchEvent(new InputEvent("input", { bubbles: !0 }));
      };
      this.ui.telInput.addEventListener("paste", e, {
        signal: this.abortController.signal
      });
    }
  }
  //* Adhere to the input's maxlength attr.
  _cap(e) {
    const n = Number(this.ui.telInput.getAttribute("maxlength"));
    return n && e.length > n ? e.substring(0, n) : e;
  }
  //* Trigger a custom event on the input (typed via ItiEventMap).
  _trigger(e, n = {}) {
    const i = new CustomEvent(e, {
      bubbles: !0,
      cancelable: !0,
      detail: n
    });
    this.ui.telInput.dispatchEvent(i);
  }
  //* Open the dropdown.
  _openDropdown() {
    const { fixDropdownWidth: e, countrySearch: n } = this.options;
    if (this.dropdownAbortController = new AbortController(), e && (this.ui.dropdownContent.style.width = `${this.ui.telInput.offsetWidth}px`), this.ui.dropdownContent.classList.remove(I.HIDE), this.ui.selectedCountry.setAttribute(N.EXPANDED, "true"), this._setDropdownPosition(), n) {
      const i = this.ui.countryList.firstElementChild;
      i && (this.ui.highlightListItem(i, !1), this.ui.countryList.scrollTop = 0), this.options.dropdownAlwaysOpen || this.ui.searchInput.focus();
    }
    this._bindDropdownListeners(), this.ui.dropdownArrow.classList.add(I.ARROW_UP), this._trigger(Gt.OPEN_COUNTRY_DROPDOWN);
  }
  //* Set the dropdown position
  _setDropdownPosition() {
    if (this.options.dropdownContainer && this.options.dropdownContainer.appendChild(this.ui.dropdown), !this.options.useFullscreenPopup) {
      const e = this.ui.telInput.getBoundingClientRect(), n = this.ui.telInput.offsetHeight;
      if (this.options.dropdownContainer) {
        this.ui.dropdown.style.top = `${e.top + n}px`, this.ui.dropdown.style.left = `${e.left}px`;
        const i = () => this._closeDropdown();
        window.addEventListener("scroll", i, {
          signal: this.dropdownAbortController.signal
        });
      }
    }
  }
  //* We only bind dropdown listeners when the dropdown is open.
  _bindDropdownListeners() {
    const e = this.dropdownAbortController.signal;
    this._bindDropdownMouseoverListener(e), this._bindDropdownCountryClickListener(e), this.options.dropdownAlwaysOpen || this._bindDropdownClickOffListener(e), this._bindDropdownKeydownListener(e), this.options.countrySearch && this._bindDropdownSearchListeners(e);
  }
  //* When mouse over a list item, just highlight that one
  //* we add the class "highlight", so if they hit "enter" we know which one to select.
  _bindDropdownMouseoverListener(e) {
    const n = (i) => {
      var r;
      const s = (r = i.target) == null ? void 0 : r.closest(
        `.${I.COUNTRY_ITEM}`
      );
      s && this.ui.highlightListItem(s, !1);
    };
    this.ui.countryList.addEventListener(
      "mouseover",
      n,
      {
        signal: e
      }
    );
  }
  //* Listen for country selection.
  _bindDropdownCountryClickListener(e) {
    const n = (i) => {
      var r;
      const s = (r = i.target) == null ? void 0 : r.closest(
        `.${I.COUNTRY_ITEM}`
      );
      s && this._selectListItem(s);
    };
    this.ui.countryList.addEventListener("click", n, {
      signal: e
    });
  }
  //* Click off to close (except when this initial opening click is bubbling up).
  //* We cannot just stopPropagation as it may be needed to close another instance.
  _bindDropdownClickOffListener(e) {
    const n = (i) => {
      !!i.target.closest(
        `#iti-${this.id}__dropdown-content`
      ) || this._closeDropdown();
    };
    setTimeout(() => {
      document.documentElement.addEventListener(
        "click",
        n,
        { signal: e }
      );
    }, 0);
  }
  //* Listen for up/down scrolling, enter to select, or escape to close.
  //* Use keydown as keypress doesn't fire for non-char keys and we want to catch if they
  //* just hit down and hold it to scroll down (no keyup event).
  //* Listen on the document because that's where key events are triggered if no input has focus.
  _bindDropdownKeydownListener(e) {
    let n = "", i = null;
    const s = (r) => {
      [
        M.ARROW_UP,
        M.ARROW_DOWN,
        M.ENTER,
        M.ESC
      ].includes(r.key) && (r.preventDefault(), r.stopPropagation(), r.key === M.ARROW_UP || r.key === M.ARROW_DOWN ? this._handleUpDownKey(r.key) : r.key === M.ENTER ? this._handleEnterKey() : r.key === M.ESC && (this._closeDropdown(), this.ui.selectedCountry.focus())), !this.options.countrySearch && ut.HIDDEN_SEARCH_CHAR.test(r.key) && (r.stopPropagation(), i && clearTimeout(i), n += r.key.toLowerCase(), this._searchForCountry(n), i = setTimeout(() => {
        n = "";
      }, Ps.HIDDEN_SEARCH_RESET_MS));
    };
    document.addEventListener("keydown", s, { signal: e });
  }
  //* Search input listeners when countrySearch enabled.
  _bindDropdownSearchListeners(e) {
    const n = () => {
      const o = this.ui.searchInput.value.trim();
      this._filterCountriesByQuery(o), this.ui.searchInput.value ? this.ui.searchClearButton.classList.remove(I.HIDE) : this.ui.searchClearButton.classList.add(I.HIDE);
    };
    let i = null;
    const s = () => {
      i && clearTimeout(i), i = setTimeout(() => {
        n(), i = null;
      }, 100);
    };
    this.ui.searchInput.addEventListener("input", s, {
      signal: e
    });
    const r = () => {
      this.ui.searchInput.value = "", this.ui.searchInput.focus(), n();
    };
    this.ui.searchClearButton.addEventListener("click", r, {
      signal: e
    });
  }
  //* Hidden search (countrySearch disabled): Find the first list item whose name starts with the query string.
  _searchForCountry(e) {
    const n = Gs(this.countries, e);
    if (n) {
      const i = n.nodeById[this.id];
      this.ui.highlightListItem(i, !1), this.ui.scrollTo(i);
    }
  }
  //* Country search: Filter the countries according to the search query.
  _filterCountriesByQuery(e) {
    let n;
    e === "" ? n = this.countries : n = js(this.countries, e), this.ui.filterCountries(n);
  }
  //* Highlight the next/prev item in the list (and ensure it is visible).
  _handleUpDownKey(e) {
    var i, s;
    let n = e === M.ARROW_UP ? (i = this.ui.highlightedItem) == null ? void 0 : i.previousElementSibling : (s = this.ui.highlightedItem) == null ? void 0 : s.nextElementSibling;
    !n && this.ui.countryList.childElementCount > 1 && (n = e === M.ARROW_UP ? this.ui.countryList.lastElementChild : this.ui.countryList.firstElementChild), n && (this.ui.scrollTo(n), this.ui.highlightListItem(n, !1));
  }
  //* Select the currently highlighted item.
  _handleEnterKey() {
    this.ui.highlightedItem && this._selectListItem(this.ui.highlightedItem);
  }
  //* Update the input's value to the given val (format first if possible)
  //* NOTE: this is called from _setInitialState, handleUtils and setNumber.
  _updateValFromNumber(e) {
    let n = e;
    if (this.options.formatOnDisplay && C.utils && this.selectedCountryData) {
      const i = this.options.nationalMode || !n.startsWith("+") && !this.options.separateDialCode, { NATIONAL: s, INTERNATIONAL: r } = C.utils.numberFormat, o = i ? s : r;
      n = C.utils.formatNumber(
        n,
        this.selectedCountryData.iso2,
        o
      );
    }
    n = this._beforeSetNumber(n), this._setTelInputValue(n);
  }
  //* Check if need to select a new country based on the given number
  //* Note: called from _setInitialState, keyup handler, setNumber.
  _updateCountryFromNumber(e) {
    const n = this._getNewCountryFromNumber(e);
    return n !== null ? this._setCountry(n) : !1;
  }
  // if there is a selected country, and the number doesn't start with a dial code, then add it
  _ensureHasDialCode(e) {
    const { dialCode: n, nationalPrefix: i } = this.selectedCountryData;
    if (e.startsWith("+") || !n)
      return e;
    const o = i && e.startsWith(i) && !this.options.separateDialCode ? e.substring(1) : e;
    return `+${n}${o}`;
  }
  //* Get the new country based on the input number, or return null if no change, or empty string if should be empty (e.g. if they type an invalid dial code).
  _getNewCountryFromNumber(e) {
    const n = e.indexOf("+");
    let i = n ? e.substring(n) : e;
    const s = this.selectedCountryData.iso2, r = this.selectedCountryData.dialCode;
    i = this._ensureHasDialCode(i);
    const o = this._getDialCode(i, !0), l = zt(i);
    if (o) {
      const a = zt(o), u = this.dialCodeToIso2Map[a];
      if (u.length === 1)
        return u[0] === s ? null : u[0];
      if (!s && this.defaultCountry && u.includes(this.defaultCountry))
        return this.defaultCountry;
      if (r === Bn.NANP && en(l))
        return null;
      const { areaCodes: d, priority: f } = this.selectedCountryData;
      if (d) {
        const y = d.map(
          (m) => `${r}${m}`
        );
        for (const m of y)
          if (l.startsWith(m))
            return null;
      }
      const _ = d && !(f === 0) && l.length > a.length, g = s && u.includes(s) && !_, b = s === u[0];
      if (!g && !b)
        return u[0];
    } else if (i.startsWith("+") && l.length) {
      const a = this.selectedCountryData.dialCode || "";
      return a && a.startsWith(l) ? null : "";
    } else if ((!i || i === "+") && !s && this.defaultCountry)
      return this.defaultCountry;
    return null;
  }
  //* Update the selected country, dial code (if separateDialCode), placeholder, title, and selected list item.
  //* Note: called from _setInitialState, _updateCountryFromNumber, _selectListItem, setCountry.
  _setCountry(e) {
    const { separateDialCode: n, showFlags: i, i18n: s, allowDropdown: r } = this.options, o = this.selectedCountryData.iso2 || "";
    if (r && this.ui.updateSelectedItem(e), this.selectedCountryData = e ? this.countryByIso2.get(e) : {}, this.selectedCountryData.iso2 && (this.defaultCountry = this.selectedCountryData.iso2), this.ui.selectedCountry) {
      const l = e && i ? `${I.FLAG} iti__${e}` : `${I.FLAG} ${I.GLOBE}`;
      let a, u, c;
      if (e) {
        const { name: d, dialCode: f } = this.selectedCountryData;
        u = d, a = s.selectedCountryAriaLabel.replace("${countryName}", d).replace("${dialCode}", `+${f}`), c = i ? "" : Je();
      } else
        u = s.noCountrySelected, a = s.noCountrySelected, c = Je();
      this.ui.selectedCountryInner.className = l, this.ui.selectedCountry.setAttribute("title", u), this.ui.selectedCountry.setAttribute(N.LABEL, a), this.ui.selectedCountryInner.innerHTML = c;
    }
    if (n) {
      const l = this.selectedCountryData.dialCode ? `+${this.selectedCountryData.dialCode}` : "";
      this.ui.selectedDialCode.textContent = l, this.ui.updateInputPadding();
    }
    return this._updatePlaceholder(), this._updateMaxLength(), o !== e;
  }
  //* Update the maximum valid number length for the currently selected country.
  _updateMaxLength() {
    const { strictMode: e, placeholderNumberType: n, allowedNumberTypes: i } = this.options, { iso2: s } = this.selectedCountryData;
    if (e && C.utils)
      if (s) {
        const r = C.utils.numberType[n];
        let o = C.utils.getExampleNumber(
          s,
          !1,
          r,
          !0
        ), l = o;
        for (; C.utils.isPossibleNumber(
          o,
          s,
          i
        ); )
          l = o, o += "0";
        const a = C.utils.getCoreNumber(l, s);
        this.maxCoreNumberLength = a.length, s === "by" && (this.maxCoreNumberLength = a.length + 1);
      } else
        this.maxCoreNumberLength = null;
  }
  //* Update the input placeholder to an example number from the currently selected country.
  _updatePlaceholder() {
    const {
      autoPlaceholder: e,
      placeholderNumberType: n,
      nationalMode: i,
      customPlaceholder: s
    } = this.options, r = e === te.AGGRESSIVE || !this.ui.hadInitialPlaceholder && e === te.POLITE;
    if (C.utils && r) {
      const o = C.utils.numberType[n];
      let l = this.selectedCountryData.iso2 ? C.utils.getExampleNumber(
        this.selectedCountryData.iso2,
        i,
        o
      ) : "";
      l = this._beforeSetNumber(l), typeof s == "function" && (l = s(l, this.selectedCountryData)), this.ui.telInput.setAttribute("placeholder", l);
    }
  }
  //* Called when the user selects a list item from the dropdown.
  _selectListItem(e) {
    const n = e.dataset[qe.COUNTRY_CODE], i = this._setCountry(n);
    this._closeDropdown();
    const s = e.dataset[qe.DIAL_CODE];
    if (this._updateDialCode(s), this.options.formatOnDisplay) {
      const r = this._getTelInputValue();
      this._updateValFromNumber(r);
    }
    this.ui.telInput.focus(), i && this._triggerCountryChange();
  }
  //* Close the dropdown and unbind any listeners.
  _closeDropdown() {
    this.ui.dropdownContent.classList.contains(I.HIDE) || this.options.dropdownAlwaysOpen || (this.ui.dropdownContent.classList.add(I.HIDE), this.ui.selectedCountry.setAttribute(N.EXPANDED, "false"), this.options.countrySearch && (this.ui.searchInput.removeAttribute(N.ACTIVE_DESCENDANT), this.ui.highlightedItem && (this.ui.highlightedItem.classList.remove(I.HIGHLIGHT), this.ui.highlightedItem = null)), this.ui.dropdownArrow.classList.remove(I.ARROW_UP), this.dropdownAbortController.abort(), this.dropdownAbortController = null, this.options.dropdownContainer && this.ui.dropdown.remove(), this._trigger(Gt.CLOSE_COUNTRY_DROPDOWN));
  }
  //* Replace any existing dial code with the new one
  //* Note: called from _selectListItem and setCountry
  _updateDialCode(e) {
    const n = this._getTelInputValue(), i = `+${e}`;
    let s;
    if (n.startsWith("+")) {
      const r = this._getDialCode(n);
      r ? s = n.replace(r, i) : s = i, this._setTelInputValue(s);
    }
  }
  //* Try and extract a valid international dial code from a full telephone number.
  //* Note: returns the raw string inc plus character and any whitespace/dots etc.
  _getDialCode(e, n) {
    let i = "";
    if (e.startsWith("+")) {
      let s = "", r = !1;
      for (let o = 0; o < e.length; o++) {
        const l = e.charAt(o);
        if (/[0-9]/.test(l)) {
          if (s += l, !!!this.dialCodeToIso2Map[s])
            break;
          if (this.dialCodes.has(s)) {
            if (i = e.substring(0, o + 1), r = !0, !n)
              break;
          } else n && r && (i = e.substring(0, o + 1));
          if (s.length === this.dialCodeMaxLen)
            break;
        }
      }
    }
    return i;
  }
  //* Get the input val, adding the dial code if separateDialCode is enabled.
  _getFullNumber(e) {
    const n = e ? this._normaliseNumerals(e) : this._getTelInputValue(), { dialCode: i } = this.selectedCountryData;
    let s;
    const r = zt(n);
    return this.options.separateDialCode && !n.startsWith("+") && i && r ? s = `+${i}` : s = "", s + n;
  }
  //* Remove the dial code if separateDialCode is enabled also cap the length if the input has a maxlength attribute
  _beforeSetNumber(e) {
    const n = !!this._getDialCode(e), i = tn(
      e,
      n,
      this.options.separateDialCode,
      this.selectedCountryData
    );
    return this._cap(i);
  }
  //* Trigger the 'countrychange' event.
  _triggerCountryChange() {
    this._trigger(Gt.COUNTRY_CHANGE);
  }
  //**************************
  //*  SECRET PUBLIC METHODS
  //**************************
  //* This is called when the geoip call returns.
  handleAutoCountry() {
    this.options.initialCountry === Mt.AUTO && C.autoCountry && (this.defaultCountry = C.autoCountry, this.selectedCountryData.iso2 || this.ui.selectedCountryInner.classList.contains(I.GLOBE) || this.setCountry(this.defaultCountry), this.resolveAutoCountryPromise());
  }
  //* This is called when the geoip call fails or times out.
  handleAutoCountryFailure() {
    this._setInitialState(!0), this.rejectAutoCountryPromise();
  }
  //* This is called when the utils request completes.
  handleUtils() {
    if (C.utils) {
      const e = this._getTelInputValue();
      e && this._updateValFromNumber(e), this.selectedCountryData.iso2 && (this._updatePlaceholder(), this._updateMaxLength());
    }
    this.resolveUtilsScriptPromise();
  }
  //* This is called when the utils request fails or times out.
  handleUtilsFailure(e) {
    this.rejectUtilsScriptPromise(e);
  }
  //********************
  //*  PUBLIC METHODS
  //********************
  //* Remove plugin.
  destroy() {
    this.ui.telInput && (this.options.allowDropdown && this._closeDropdown(), this.abortController.abort(), this.abortController = null, this.ui.destroy(), C.instances instanceof Map ? C.instances.delete(this.id) : delete C.instances[this.id]);
  }
  //* Get the extension from the current number.
  getExtension() {
    return C.utils ? C.utils.getExtension(
      this._getFullNumber(),
      this.selectedCountryData.iso2
    ) : "";
  }
  //* Format the number to the given format.
  getNumber(e) {
    if (C.utils) {
      const { iso2: n } = this.selectedCountryData, i = this._getFullNumber(), s = C.utils.formatNumber(
        i,
        n,
        e
      );
      return this._mapAsciiToUserNumerals(s);
    }
    return "";
  }
  //* Get the type of the entered number e.g. landline/mobile.
  getNumberType() {
    return C.utils ? C.utils.getNumberType(
      this._getFullNumber(),
      this.selectedCountryData.iso2
    ) : ze.UNKNOWN_NUMBER_TYPE;
  }
  //* Get the country data for the currently selected country.
  getSelectedCountryData() {
    return this.selectedCountryData;
  }
  //* Get the validation error.
  getValidationError() {
    if (C.utils) {
      const { iso2: e } = this.selectedCountryData;
      return C.utils.getValidationError(this._getFullNumber(), e);
    }
    return ze.UNKNOWN_VALIDATION_ERROR;
  }
  //* Validate the input val using number length only
  isValidNumber() {
    const { dialCode: e, iso2: n } = this.selectedCountryData;
    if (e === ae.DIAL_CODE && C.utils) {
      const i = this._getFullNumber(), s = C.utils.getCoreNumber(i, n);
      if (s[0] === ae.MOBILE_PREFIX && s.length !== ae.MOBILE_CORE_LENGTH)
        return !1;
    }
    return this._validateNumber(!1);
  }
  //* Validate the input val with precise validation
  isValidNumberPrecise() {
    return this._validateNumber(!0);
  }
  _utilsIsPossibleNumber(e) {
    return C.utils ? C.utils.isPossibleNumber(
      e,
      this.selectedCountryData.iso2,
      this.options.allowedNumberTypes
    ) : null;
  }
  //* Shared internal validation logic to handle alpha character extension rules.
  _validateNumber(e) {
    if (!C.utils)
      return null;
    if (!this.selectedCountryData.iso2)
      return !1;
    const { allowNumberExtensions: n, allowPhonewords: i } = this.options, s = (a) => e ? this._utilsIsValidNumber(a) : this._utilsIsPossibleNumber(a), r = this._getFullNumber();
    if (!s(r))
      return !1;
    if (r.search(ut.ALPHA_UNICODE) > -1) {
      const a = this.selectedCountryData.iso2;
      return C.utils.getExtension(r, a) ? n : i;
    }
    return !0;
  }
  _utilsIsValidNumber(e) {
    return C.utils ? C.utils.isValidNumber(
      e,
      this.selectedCountryData.iso2,
      this.options.allowedNumberTypes
    ) : null;
  }
  //* Update the selected country, and update the input val accordingly.
  setCountry(e) {
    const n = e == null ? void 0 : e.toLowerCase();
    if (!de(n))
      throw new Error(`Invalid country code: '${n}'`);
    const i = this.selectedCountryData.iso2;
    if (e && n !== i || !e && i) {
      if (this._setCountry(n), this._updateDialCode(this.selectedCountryData.dialCode), this.options.formatOnDisplay) {
        const r = this._getTelInputValue();
        this._updateValFromNumber(r);
      }
      this._triggerCountryChange();
    }
  }
  //* Set the input value and update the country.
  setNumber(e) {
    const n = this._normaliseNumerals(e), i = this._updateCountryFromNumber(n);
    this._updateValFromNumber(n), i && this._triggerCountryChange(), this._trigger(Gt.INPUT, { isSetNumber: !0 });
  }
  //* Set the placeholder number typ
  setPlaceholderNumberType(e) {
    this.options.placeholderNumberType = e, this._updatePlaceholder();
  }
  setDisabled(e) {
    this.ui.telInput.disabled = e, e ? this.ui.selectedCountry.setAttribute("disabled", "true") : this.ui.selectedCountry.removeAttribute("disabled");
  }
}
const lr = (t) => {
  if (!C.utils && !C.startedLoadingUtilsScript) {
    let e;
    if (typeof t == "function")
      try {
        e = Promise.resolve(t());
      } catch (n) {
        return Promise.reject(n);
      }
    else
      return Promise.reject(
        new TypeError(
          `The argument passed to attachUtils must be a function that returns a promise for the utilities module, not ${typeof t}`
        )
      );
    return C.startedLoadingUtilsScript = !0, e.then((n) => {
      const i = n == null ? void 0 : n.default;
      if (!i || typeof i != "object")
        throw new TypeError(
          "The loader function passed to attachUtils did not resolve to a module object with utils as its default export."
        );
      return C.utils = i, Ut("handleUtils"), !0;
    }).catch((n) => {
      throw Ut("handleUtilsFailure", n), n;
    });
  }
  return null;
}, Ut = (t, ...e) => {
  Object.values(C.instances).forEach((n) => {
    const i = n[t];
    typeof i == "function" && i.apply(n, e);
  });
}, C = Object.assign(
  (t, e) => {
    const n = new or(t, e);
    return C.instances[n.id] = n, t.iti = n, n;
  },
  {
    defaults: Pe,
    //* Using a static var like this allows us to mock it in the tests.
    documentReady: () => document.readyState === "complete",
    //* Get the country data object.
    getCountryData: () => pt,
    //* A getter for the plugin instance.
    getInstance: (t) => {
      const e = t.dataset.intlTelInputId;
      return e ? C.instances[e] : null;
    },
    //* A map from instance ID to instance object.
    instances: {},
    attachUtils: lr,
    startedLoadingUtilsScript: !1,
    startedLoadingAutoCountry: !1,
    version: "26.4.0"
  }
);
var ar = /* @__PURE__ */ cs("<input/>");
function fr(t, e) {
  Di(e, !0);
  let n = jt(e, "disabled", 3, !1), i = jt(e, "inputProps", 19, () => ({})), s = jt(e, "options", 19, () => ({})), r = jt(e, "value", 3, ""), o = /* @__PURE__ */ Y(void 0), l = /* @__PURE__ */ Y(void 0), a = /* @__PURE__ */ Y(null), u = /* @__PURE__ */ Y(!1);
  const c = () => E(l) ? s().strictMode ? E(l).isValidNumberPrecise() : E(l).isValidNumber() : null, d = () => {
    var p, v;
    const m = c();
    E(a) !== m && (W(a, m, !0), (p = e.onChangeValidity) == null || p.call(e, !!m), (v = e.onChangeErrorCode) == null || v.call(e, m ? null : E(l).getValidationError()));
  }, f = () => {
    var m, p;
    (p = e.onChangeNumber) == null || p.call(e, ((m = E(l)) == null ? void 0 : m.getNumber()) ?? ""), d();
  }, h = () => {
    var m, p;
    (p = e.onChangeCountry) == null || p.call(e, ((m = E(l)) == null ? void 0 : m.getSelectedCountryData().iso2) ?? ""), f();
  };
  zn(() => {
    E(o) && (W(l, C(E(o), s()), !0), E(o).addEventListener("countrychange", h), r() && E(l).setNumber(r()), n() && E(l).setDisabled(n()), W(a, c(), !0), W(u, !0));
  }), qn(() => {
    var m;
    E(o) && E(o).removeEventListener("countrychange", h), (m = E(l)) == null || m.destroy();
  }), qi(() => {
    E(u) && E(l) && E(l).setDisabled(n());
  });
  function _() {
    return E(l);
  }
  function g() {
    return E(o);
  }
  var b = { getInstance: _, getInput: g }, y = ar();
  return Ls(y, () => ({ type: "tel", oninput: f, ...i() }), void 0, void 0, void 0, void 0, !0), As(y, (m) => W(o, m), () => E(o)), ds(t, y), Li(b);
}
export {
  fr as default
};
