// intl-tel-input/vue/build/exports/IntlTelInput.mjs
import { defineComponent as h, computed as C, ref as s, onMounted as N, watch as y, onUnmounted as P, openBlock as _, createElementBlock as x, mergeProps as B } from "vue";
import { i as E } from "../intl-tel-input-9R5re0vW.mjs";
const I = ["value"], w = /* @__PURE__ */ h({
  inheritAttrs: !1,
  __name: "IntlTelInput",
  props: {
    options: { default: () => ({}) },
    usePreciseValidation: { type: Boolean, default: !1 },
    disabled: { type: Boolean, default: !1 },
    inputProps: { default: () => ({}) },
    value: {},
    modelValue: {}
  },
  emits: [
    "changeNumber",
    "changeCountry",
    "changeValidity",
    "changeErrorCode",
    "update:modelValue"
  ],
  setup(d, { expose: V, emit: b }) {
    const l = d, u = b, n = C(() => l.value ?? l.modelValue ?? ""), o = s(null), t = s(null), i = s(!1), c = () => t.value ? l.usePreciseValidation ? t.value.isValidNumberPrecise() : t.value.isValidNumber() : null, p = () => {
      var a, r;
      let e = c();
      i.value !== e && (i.value = e, u("changeValidity", !!e), u(
        "changeErrorCode",
        e ? null : ((r = (a = t.value) == null ? void 0 : a.getValidationError) == null ? void 0 : r.call(a)) ?? null
      ));
    }, v = () => {
      var a;
      const e = ((a = t.value) == null ? void 0 : a.getNumber()) ?? "";
      u("changeNumber", e), u("update:modelValue", e), p();
    }, g = () => {
      var e;
      u("changeCountry", ((e = t.value) == null ? void 0 : e.getSelectedCountryData().iso2) ?? ""), v();
    };
    return N(() => {
      o.value && (t.value = E(o.value, l.options), n.value && t.value.setNumber(n.value), l.disabled && t.value.setDisabled(l.disabled), i.value = c());
    }), y(
      () => l.disabled,
      (e) => {
        var a;
        return (a = t.value) == null ? void 0 : a.setDisabled(e);
      }
    ), y(
      () => n.value,
      (e) => {
        var m, f;
        if (!t.value) return;
        const a = e ?? "";
        (((f = (m = t.value).getNumber) == null ? void 0 : f.call(m)) ?? "") !== a && (t.value.setNumber(a), p());
      },
      { flush: "post" }
    ), P(() => {
      var e;
      return (e = t.value) == null ? void 0 : e.destroy();
    }), V({ instance: t, input: o }), (e, a) => (_(), x("input", B({
      ref_key: "input",
      ref: o,
      value: n.value,
      type: "tel",
      onCountrychange: g,
      onInput: v
    }, d.inputProps), null, 16, I));
  }
});
export {
  w as default
};
